<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Omar Johnson" />

<meta name="date" content="2024-07-16" />

<title>Johnson_DOX_24_2</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Johnson_DOX_24</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Johnson_DOX_24_2</h1>
<h4 class="author">Omar Johnson</h4>
<h4 class="date">2024-07-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DOX_24_Github/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240723code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240723)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240723code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240723)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong88c6686">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 88c6686
</a>
</p>
</div>
<div id="strongRepositoryversionstrong88c6686"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 88c6686.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/Johnson_DOX_24_2.html
    Ignored:    analysis/Johnson_DOX_24_3.html
    Ignored:    analysis/Johnson_DOX_24_4.html
    Ignored:    analysis/Johnson_DOX_24_5.html
    Ignored:    analysis/Johnson_DOX_24_6.html
    Ignored:    analysis/Johnson_DOX_24_7.html
    Ignored:    analysis/Johnson_DOX_24_8.html
    Ignored:    analysis/Johnson_DOX_24_RUV_Limma.html

Untracked files:
    Untracked:  Fig_2.Rmd
    Untracked:  Fig_2.html
    Untracked:  analysis/Johnson_DOX_24_RUV_Limma.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Johnson_DOX_24_2.Rmd</code>) and
HTML (<code>docs/Johnson_DOX_24_2.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
88c6686
</td>
<td>
Omar-Johnson
</td>
<td>
2024-07-23
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
</div>
<div id="read-in-data" class="section level1">
<h1>Read in Data</h1>
</div>
<div id="generate-and-view-initial-objects" class="section level1">
<h1>Generate and view initial objects</h1>
<pre class="r"><code># Undo log tranformation
RUVg_Log2_quantnormalized_all10samples %&gt;% head()</code></pre>
<pre><code>                 S1       S3       S5       S7       S9       S2       S4
A0A0B4J2A2 328.5554 326.9973 344.8820 349.3119 352.7209 333.5604 343.2972
A0A0B4J2D5 575.3541 580.4938 574.6771 568.2429 573.3663 571.9707 564.5982
A0A494C071 319.5122 317.2598 303.3583 322.1064 317.5238 328.8213 328.9263
A0AVT1     174.4011 178.2046 205.8086 196.3262 195.1595 173.2376 180.2296
A0FGR8     411.5794 406.2913 397.4114 401.4574 400.3695 405.4398 406.3051
A0JLT2     324.3169 318.3035 315.9126 304.1986 322.8292 324.8402 324.0475
                 S6       S8      S10
A0A0B4J2A2 343.7512 340.3927 346.0124
A0A0B4J2D5 564.3974 562.3821 559.3047
A0A494C071 316.6769 326.8686 312.6095
A0AVT1     195.4427 178.4695 182.9383
A0FGR8     403.6977 405.8293 402.7064
A0JLT2     309.3706 297.4752 323.7055</code></pre>
<pre class="r"><code>RUVg_Log2_quantnormalized_all10samples %&gt;% dim()</code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code># Create summarized experiment object 
WGCNA_DF &lt;- SummarizedExperiment(assays = as.matrix(RUVg_Log2_quantnormalized_all10samples))
dim(RUVg_Log2_quantnormalized_all10samples) </code></pre>
<pre><code>[1] 4178   10</code></pre>
<pre class="r"><code>exp_filt &lt;- filter_by_variance(WGCNA_DF, n = 4178)
exp_filt %&gt;% assay() %&gt;% head()</code></pre>
<pre><code>                 S1       S3       S5       S7       S9       S2       S4
A0A0B4J2A2 328.5554 326.9973 344.8820 349.3119 352.7209 333.5604 343.2972
A0A0B4J2D5 575.3541 580.4938 574.6771 568.2429 573.3663 571.9707 564.5982
A0A494C071 319.5122 317.2598 303.3583 322.1064 317.5238 328.8213 328.9263
A0AVT1     174.4011 178.2046 205.8086 196.3262 195.1595 173.2376 180.2296
A0FGR8     411.5794 406.2913 397.4114 401.4574 400.3695 405.4398 406.3051
A0JLT2     324.3169 318.3035 315.9126 304.1986 322.8292 324.8402 324.0475
                 S6       S8      S10
A0A0B4J2A2 343.7512 340.3927 346.0124
A0A0B4J2D5 564.3974 562.3821 559.3047
A0A494C071 316.6769 326.8686 312.6095
A0AVT1     195.4427 178.4695 182.9383
A0FGR8     403.6977 405.8293 402.7064
A0JLT2     309.3706 297.4752 323.7055</code></pre>
<pre class="r"><code>toptable_summary %&gt;% head()</code></pre>
<pre><code>           logFC  AveExpr         t      P.Value    adj.P.Val        B Protein
Q15061  2.436903 18.60721  26.00939 1.499311e-07 1.499311e-07 7.414993  Q15061
O76021 -2.224860 21.98800 -18.86036 1.191879e-06 1.191879e-06 6.029757  O76021
P43694 -1.821642 18.81578 -15.98230 3.129665e-06 3.129665e-06 5.276514  P43694
P01130 -1.819689 20.01194 -15.77595 3.483331e-06 3.483331e-06 5.186033  P01130
Q96P11 -1.843458 17.28018 -14.33937 5.735316e-06 5.735316e-06 4.764600  Q96P11
Q6EMK4  1.838032 19.30419  14.22737 6.455771e-06 6.455771e-06 4.660093  Q6EMK4
       threshold_P
Q15061        TRUE
O76021        TRUE
P43694        TRUE
P01130        TRUE
Q96P11        TRUE
Q6EMK4        TRUE</code></pre>
<pre class="r"><code>Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
</div>
<div id="s3-ab-sft-power-threshold" class="section level1">
<h1>S3-A&amp;B-SFT power threshold</h1>
<pre class="r"><code># Determine the soft power threshold to get scale free shape
sft &lt;- SFT_fit(exp_filt, net_type = &quot;signed&quot;, rsquared = 0.7,  cor_method = &quot;pearson&quot;)</code></pre>
<pre><code>Warning: executing %dopar% sequentially: no parallel backend registered</code></pre>
<pre><code>   Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
1      3  0.55000  2.6800          0.956   929.0     932.0   1260
2      4  0.57900  1.9100          0.964   683.0     681.0   1000
3      5  0.49000  1.2600          0.947   526.0     520.0    832
4      6  0.34100  0.7670          0.917   418.0     408.0    712
5      7  0.17700  0.4390          0.860   340.0     331.0    620
6      8  0.02530  0.1350          0.806   283.0     273.0    551
7      9  0.00557 -0.0588          0.776   239.0     229.0    494
8     10  0.09410 -0.2400          0.775   204.0     194.0    448
9     11  0.21900 -0.3780          0.792   177.0     166.0    408
10    12  0.34300 -0.4890          0.820   154.0     143.0    374
11    13  0.43300 -0.5810          0.832   136.0     124.0    345
12    14  0.50200 -0.6480          0.847   120.0     109.0    319
13    15  0.55400 -0.7050          0.866   107.0      95.7    297
14    16  0.59700 -0.7490          0.883    96.3      84.6    277
15    17  0.64100 -0.7900          0.902    86.9      75.3    259
16    18  0.66900 -0.8330          0.909    78.7      67.1    243
17    19  0.69800 -0.8680          0.920    71.6      59.8    228
18    20  0.70900 -0.9210          0.916    65.4      53.6    216</code></pre>
<pre class="r"><code>sft$plot</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/SFT%20power%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="modify-workflow-to-construct-a-simpler-network"
class="section level1">
<h1>Modify workflow to construct a simpler network</h1>
<pre class="r"><code>exp2gcn2_EDIT &lt;- function (exp, net_type = &quot;signed&quot;, module_merging_threshold = 0.8, 
          SFTpower = NULL, cor_method = &quot;spearman&quot;, verbose = FALSE) 
{
  params &lt;- list(net_type = net_type, module_merging_threshold = module_merging_threshold, 
                 SFTpower = SFTpower, cor_method = cor_method)
  norm.exp &lt;- BioNERO:::handleSE(exp)
  if (is.null(SFTpower)) {
    stop(&quot;Please, specify the SFT power.&quot;)
  }
  if (verbose) {
    message(&quot;Calculating adjacency matrix...&quot;)
  }
  cor_matrix &lt;- BioNERO:::calculate_cor_adj(cor_method, norm.exp, SFTpower, 
                                  net_type)$cor
  adj_matrix &lt;- BioNERO:::calculate_cor_adj(cor_method, norm.exp, SFTpower, 
                                  net_type)$adj
  gene_ids &lt;- rownames(adj_matrix)
  adj_matrix &lt;- matrix(adj_matrix, nrow = nrow(adj_matrix))
  rownames(adj_matrix) &lt;- gene_ids
  colnames(adj_matrix) &lt;- gene_ids
  if (verbose) {
    message(&quot;Calculating topological overlap matrix (TOM)...&quot;)
  }
  tomtype &lt;- BioNERO:::get_TOMtype(net_type)
  TOM &lt;- WGCNA::TOMsimilarity(adj_matrix, TOMType = tomtype)
  dissTOM &lt;- 1 - TOM
  geneTree &lt;- hclust(as.dist(dissTOM), method = &quot;average&quot;)
  if (verbose) {
    message(&quot;Detecting coexpression modules...&quot;)
  }
  old.module_labels &lt;- dynamicTreeCut::cutreeDynamicTree(dendro = geneTree, maxTreeHeight = 3 , 
                                                         minModuleSize = 40, deepSplit = FALSE)
  nmod &lt;- length(unique(old.module_labels))
  palette &lt;- rev(WGCNA::standardColors(nmod))
  old.module_colors &lt;- WGCNA::labels2colors(old.module_labels, 
                                            colorSeq = palette)
  if (verbose) {
    message(&quot;Calculating module eigengenes (MEs)...&quot;)
  }
  old.MElist &lt;- WGCNA::moduleEigengenes(t(norm.exp), colors = old.module_colors, 
                                        softPower = SFTpower)
  old.MEs &lt;- old.MElist$eigengenes
  MEDiss1 &lt;- 1 - cor(old.MEs)
  old.METree &lt;- hclust(as.dist(MEDiss1), method = &quot;average&quot;)
  MEDissThreshold &lt;- 1 - module_merging_threshold
  if (verbose) {
    message(&quot;Merging similar modules...&quot;)
  }
  if (cor_method == &quot;pearson&quot;) {
    merge1 &lt;- WGCNA::mergeCloseModules(t(norm.exp), old.module_colors, 
                                       cutHeight = MEDissThreshold, verbose = 0, colorSeq = palette)
  }
  else if (cor_method == &quot;spearman&quot;) {
    merge1 &lt;- WGCNA::mergeCloseModules(t(norm.exp), old.module_colors, 
                                       cutHeight = MEDissThreshold, verbose = 0, corOptions = list(use = &quot;p&quot;, 
                                                                                                   method = &quot;spearman&quot;), colorSeq = palette)
  }
  else if (cor_method == &quot;biweight&quot;) {
    merge1 &lt;- WGCNA::mergeCloseModules(t(norm.exp), old.module_colors, 
                                       cutHeight = MEDissThreshold, verbose = 0, corFnc = bicor, 
                                       colorSeq = palette)
  }
  else {
    stop(&quot;Please, specify a correlation method. One of &#39;spearman&#39;, &#39;pearson&#39; or &#39;biweight&#39;.&quot;)
  }
  new.module_colors &lt;- merge1$colors
  new.MEs &lt;- merge1$newMEs
  new.METree &lt;- hclust(as.dist(1 - cor(new.MEs)), method = &quot;average&quot;)
  genes_and_modules &lt;- as.data.frame(cbind(gene_ids, new.module_colors))
  colnames(genes_and_modules) &lt;- c(&quot;Genes&quot;, &quot;Modules&quot;)
  if (verbose) {
    message(&quot;Calculating intramodular connectivity...&quot;)
  }
  kwithin &lt;- WGCNA::intramodularConnectivity(adj_matrix, new.module_colors)
  result.list &lt;- list(adjacency_matrix = adj_matrix, MEs = new.MEs, 
                      genes_and_modules = genes_and_modules, kIN = kwithin, 
                      correlation_matrix = cor_matrix, params = params, dendro_plot_objects = list(tree = geneTree, 
                                                                                                   unmerged = old.module_colors))
  return(result.list)
}</code></pre>
</div>
<div id="fig-s3-cd" class="section level1">
<h1>Fig S3-C,D</h1>
<pre class="r"><code>net &lt;- exp2gcn2_EDIT(
  exp_filt, net_type = &quot;signed&quot;, module_merging_threshold = 0.9999, SFTpower = 20, 
  cor_method = &quot;pearson&quot;)</code></pre>
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
<pre class="r"><code># Eigengene networks
WGCNA::plotEigengeneNetworks(net$MEs, &quot;&quot;, marDendro = c(3, 5, 2, 6), plotHeatmaps = FALSE)</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20max%20merge%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>WGCNA::plotEigengeneNetworks(net$MEs, &quot;&quot;, marHeatmap = c(3, 4, 2, 2), plotDendrograms = FALSE)</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20max%20merge%20threshold-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Genes per module 
plot_ngenes_per_module(net)</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20max%20merge%20threshold-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-a-fig-s3-e-weighted-protein-co-expression-network"
class="section level1">
<h1>Figure-2-A &amp; Fig S3-E Weighted protein co-expression
network</h1>
<pre class="r"><code>net &lt;- exp2gcn2_EDIT(
  exp_filt, net_type = &quot;signed&quot;, module_merging_threshold = 0.85, SFTpower = 20, 
  cor_method = &quot;pearson&quot;)</code></pre>
<pre><code>..connectivity..
..matrix multiplication (system BLAS)..
..normalization..
..done.</code></pre>
<pre class="r"><code># Dendogram
plot_dendro_and_colors(net )</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
<pre class="r"><code># Eigengene networks
WGCNA::plotEigengeneNetworks(net$MEs, &quot;&quot;, marDendro = c(3, 5, 2, 6), plotHeatmaps = FALSE, excludeGrey = TRUE)</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>WGCNA::plotEigengeneNetworks(net$MEs, &quot;&quot;, marHeatmap = c(3, 4, 2, 2), plotDendrograms = FALSE)</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Genes per module 
plot_ngenes_per_module(net)</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Generate%20network%20with%20desired%20merge%20threshold-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fig-2-a-define-trait-and-get-eigen-proteins"
class="section level1">
<h1>Fig-2-A-Define trait and get eigen proteins</h1>
<pre class="r"><code>Meta</code></pre>
<pre><code>    Samples  Ind  Rep    Cond     Cond_Ind
S1       S1 B_77  Bio     Dox     B_77_Dox
S3       S3 C_87  Bio     Dox     C_87_Dox
S5       S5 A_48 Tech     Dox     A_48_Dox
S7       S7 A_48 Tech     Dox     A_48_Dox
S9       S9 A_48 Tech     Dox     A_48_Dox
S2       S2 B_77  Bio Control B_77_Control
S4       S4 C_87  Bio Control C_87_Control
S6       S6 A_48 Tech Control A_48_Control
S8       S8 A_48 Tech Control A_48_Control
S10     S10 A_48 Tech Control A_48_Control</code></pre>
<pre class="r"><code>WGCNA_DF$trait &lt;- c(1,1,1,1,1,0,0,0,0,0)
names(WGCNA_DF$trait) &lt;- c(&quot;S1&quot;, &quot;S3&quot;, &quot;S5&quot;, &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot;, &quot;S8&quot;, &quot;S10&quot;)


eigenmatrix &lt;- net$MEs %&gt;% as.matrix()
eigenmatrix</code></pre>
<pre><code>        MEbrown  MEdarkred   MEmagenta MElightgreen     MEblue    MEgreen
S1   0.36003677 -0.2639588 -0.53400488    0.1624083  0.3173739 -0.1131196
S3   0.43134287 -0.2307183 -0.32791330    0.0133864 -0.3246717 -0.3061329
S5  -0.27190088  0.3619056  0.07758495   -0.5740907 -0.2186320 -0.3565624
S7  -0.06510474  0.2583481  0.16556089   -0.5684963 -0.1725624 -0.3424670
S9  -0.10247250  0.3147035  0.14165243    0.2031542 -0.3216840 -0.4130655
S2  -0.32744358 -0.5916179 -0.32584468   -0.0257309 -0.3667293  0.2538724
S4  -0.56198938 -0.3694736 -0.24572255   -0.1004831  0.1181946  0.3625280
S6  -0.03609935  0.1421290  0.23242445    0.3285150  0.1890285  0.3088566
S8   0.34498267  0.1584062  0.40111124    0.2408316  0.1401821  0.2612112
S10  0.22864811  0.2202760  0.41515145    0.3205055  0.6395003  0.3448792
    MElightyellow    MEsalmon    MEyellow MEroyalblue MEdarkgreen
S1    -0.42213716 -0.20861515 -0.16354103   0.6227290   0.2200339
S3    -0.43037879 -0.21728776 -0.09059440   0.2908742   0.0829352
S5    -0.21202797 -0.59149983  0.03250180  -0.2830170   0.4863249
S7    -0.08734748 -0.19344921 -0.06042544  -0.2845027   0.3021103
S9    -0.10372050 -0.07672494  0.01599754  -0.2498493   0.3220647
S2    -0.08510748  0.07920073  0.12616898   0.2954940  -0.5532726
S4     0.06259889  0.05635206  0.13976425   0.2536810  -0.3437305
S6     0.33150327  0.17724855  0.36410610  -0.1669567  -0.1742458
S8     0.42785242  0.47947091 -0.78440804  -0.3104020  -0.1741527
S10    0.51876480  0.49530464  0.42043023  -0.1680506  -0.1680674
    MEmidnightblue      MEgrey
S1      0.45236688 -0.19623808
S3      0.31111504  0.12600955
S5      0.21889338  0.51048881
S7      0.19025192  0.14926146
S9      0.18317149  0.42428210
S2     -0.12746118 -0.42289741
S4     -0.07461793 -0.54531562
S6     -0.15044641 -0.06538022
S8     -0.63044435  0.04181154
S10    -0.37282884 -0.02202211</code></pre>
<pre class="r"><code>correlationResults &lt;- cor(eigenmatrix, WGCNA_DF$trait, method= &quot;pearson&quot;)

pvalues &lt;- sapply(1:ncol(eigenmatrix), function(column) cor.test(eigenmatrix[,column], WGCNA_DF$trait)$p.value )

WGCNA_DF$trait &lt;- as.matrix(WGCNA_DF$trait)

Modeigen &lt;- as.matrix(net$MEs)


# Trait data frame
trait_data &lt;- data.frame(
  ME = correlationResults %&gt;% rownames(),
  Value = correlationResults[,1],
  PValue = pvalues
)


trait_data</code></pre>
<pre><code>                           ME       Value       PValue
MEbrown               MEbrown  0.22256206 5.365530e-01
MEdarkred           MEdarkred  0.27845763 4.359480e-01
MEmagenta           MEmagenta -0.30175713 3.968000e-01
MElightgreen     MElightgreen -0.48296709 1.573695e-01
MEblue                 MEblue -0.45547944 1.858861e-01
MEgreen               MEgreen -0.96850913 4.141994e-06
MElightyellow   MElightyellow -0.79411869 6.080134e-03
MEsalmon             MEsalmon -0.81433512 4.127690e-03
MEyellow             MEyellow -0.16827208 6.421517e-01
MEroyalblue       MEroyalblue  0.06086385 8.673524e-01
MEdarkgreen       MEdarkgreen  0.89395628 4.859072e-04
MEmidnightblue MEmidnightblue  0.85748240 1.514177e-03
MEgrey                 MEgrey  0.64118584 4.572349e-02</code></pre>
<pre class="r"><code>Drug_trait_results &lt;- trait_data



WGCNA_DF$trait &lt;- c(1,2,3,3,3,1,2,3,3,3)
names(WGCNA_DF$trait) &lt;- c(&quot;S1&quot;, &quot;S3&quot;, &quot;S5&quot;, &quot;S7&quot;, &quot;S9&quot;, &quot;S2&quot;, &quot;S4&quot;, &quot;S6&quot;, &quot;S8&quot;, &quot;S10&quot;)


eigenmatrix &lt;- net$MEs %&gt;% as.matrix()



correlationResults &lt;- cor(eigenmatrix, WGCNA_DF$trait, method=&quot;pearson&quot;)

pvalues &lt;- sapply(1:ncol(eigenmatrix), function(column) cor.test(eigenmatrix[,column], WGCNA_DF$trait)$p.value )

WGCNA_DF$trait &lt;- as.matrix(WGCNA_DF$trait)

Modeigen &lt;- as.matrix(net$MEs)



# Trait data frame
trait_data &lt;- data.frame(
  ME = correlationResults %&gt;% rownames(),
  Value = correlationResults[,1],
  PValue = pvalues
)
trait_data$neglogP &lt;- -log(trait_data$PValue)
trait_data</code></pre>
<pre><code>                           ME       Value       PValue    neglogP
MEbrown               MEbrown  0.02587538 9.434355e-01 0.05822730
MEdarkred           MEdarkred  0.91363938 2.190321e-04 8.42629226
MEmagenta           MEmagenta  0.90652025 2.980430e-04 8.11827275
MElightgreen     MElightgreen -0.07362501 8.398155e-01 0.17457307
MEblue                 MEblue  0.12063614 7.399155e-01 0.30121933
MEgreen               MEgreen -0.13356703 7.129791e-01 0.33830321
MElightyellow   MElightyellow  0.54638987 1.022246e-01 2.28058314
MEsalmon             MEsalmon  0.16592650 6.468648e-01 0.43561802
MEyellow             MEyellow  0.01010911 9.778886e-01 0.02235954
MEroyalblue       MEroyalblue -0.94117339 4.878437e-05 9.92810051
MEdarkgreen       MEdarkgreen  0.36653672 2.975164e-01 1.21228598
MEmidnightblue MEmidnightblue -0.35034420 3.209631e-01 1.13642916
MEgrey                 MEgrey  0.65521486 3.973991e-02 3.22539939</code></pre>
<pre class="r"><code># Get modules that significantly associate with DOX treatment 
Signifigant_modules &lt;-  trait_data[trait_data$PValue &lt; 0.01, ]$ME


# Get your genes/proteins and modules DF 
Gene_mod_DF &lt;- net$genes_and_modules 


Drug_trait_results_sub &lt;- Drug_trait_results[,c(1,2,3)]
colnames(Drug_trait_results_sub) &lt;- c(&quot;Module&quot;, &quot;cor&quot;, &quot;pval&quot;)
Drug_trait_results_sub$Trait &lt;- &quot;DOX&quot;
Drug_trait_results_sub</code></pre>
<pre><code>                       Module         cor         pval Trait
MEbrown               MEbrown  0.22256206 5.365530e-01   DOX
MEdarkred           MEdarkred  0.27845763 4.359480e-01   DOX
MEmagenta           MEmagenta -0.30175713 3.968000e-01   DOX
MElightgreen     MElightgreen -0.48296709 1.573695e-01   DOX
MEblue                 MEblue -0.45547944 1.858861e-01   DOX
MEgreen               MEgreen -0.96850913 4.141994e-06   DOX
MElightyellow   MElightyellow -0.79411869 6.080134e-03   DOX
MEsalmon             MEsalmon -0.81433512 4.127690e-03   DOX
MEyellow             MEyellow -0.16827208 6.421517e-01   DOX
MEroyalblue       MEroyalblue  0.06086385 8.673524e-01   DOX
MEdarkgreen       MEdarkgreen  0.89395628 4.859072e-04   DOX
MEmidnightblue MEmidnightblue  0.85748240 1.514177e-03   DOX
MEgrey                 MEgrey  0.64118584 4.572349e-02   DOX</code></pre>
<pre class="r"><code>Ind_trait_sub &lt;- trait_data[,c(1,2,3)]
colnames(Ind_trait_sub) &lt;- c(&quot;Module&quot;,&quot;cor&quot;, &quot;pval&quot;)
Ind_trait_sub$Trait &lt;- &quot;IND.&quot;

Fulltrait_df &lt;- rbind(Drug_trait_results_sub,Ind_trait_sub)
Fulltrait_df</code></pre>
<pre><code>                        Module         cor         pval Trait
MEbrown                MEbrown  0.22256206 5.365530e-01   DOX
MEdarkred            MEdarkred  0.27845763 4.359480e-01   DOX
MEmagenta            MEmagenta -0.30175713 3.968000e-01   DOX
MElightgreen      MElightgreen -0.48296709 1.573695e-01   DOX
MEblue                  MEblue -0.45547944 1.858861e-01   DOX
MEgreen                MEgreen -0.96850913 4.141994e-06   DOX
MElightyellow    MElightyellow -0.79411869 6.080134e-03   DOX
MEsalmon              MEsalmon -0.81433512 4.127690e-03   DOX
MEyellow              MEyellow -0.16827208 6.421517e-01   DOX
MEroyalblue        MEroyalblue  0.06086385 8.673524e-01   DOX
MEdarkgreen        MEdarkgreen  0.89395628 4.859072e-04   DOX
MEmidnightblue  MEmidnightblue  0.85748240 1.514177e-03   DOX
MEgrey                  MEgrey  0.64118584 4.572349e-02   DOX
MEbrown1               MEbrown  0.02587538 9.434355e-01  IND.
MEdarkred1           MEdarkred  0.91363938 2.190321e-04  IND.
MEmagenta1           MEmagenta  0.90652025 2.980430e-04  IND.
MElightgreen1     MElightgreen -0.07362501 8.398155e-01  IND.
MEblue1                 MEblue  0.12063614 7.399155e-01  IND.
MEgreen1               MEgreen -0.13356703 7.129791e-01  IND.
MElightyellow1   MElightyellow  0.54638987 1.022246e-01  IND.
MEsalmon1             MEsalmon  0.16592650 6.468648e-01  IND.
MEyellow1             MEyellow  0.01010911 9.778886e-01  IND.
MEroyalblue1       MEroyalblue -0.94117339 4.878437e-05  IND.
MEdarkgreen1       MEdarkgreen  0.36653672 2.975164e-01  IND.
MEmidnightblue1 MEmidnightblue -0.35034420 3.209631e-01  IND.
MEgrey1                 MEgrey  0.65521486 3.973991e-02  IND.</code></pre>
<pre class="r"><code>Fulltrait_df$annotation &lt;- ifelse(Fulltrait_df$pval &lt; 0.01, &quot;*&quot;, &quot;&quot;)
module_order &lt;- c(&quot;MEbrown&quot;, &quot;MEdarkred&quot;, &quot;MEmagenta&quot;, &quot;MElightgreen&quot;, &quot;MEblue&quot;,      &quot;MEgreen&quot;,&quot;MElightyellow&quot;,&quot;MEsalmon&quot;,&quot;MEyellow&quot;,&quot;MEroyalblue&quot;,&quot;MEdarkgreen&quot;, &quot;MEmidnightblue&quot;, &quot;MEgrey&quot;)  

# Factor the Module column in Fulltrait_df
Fulltrait_df$Module &lt;- factor(Fulltrait_df$Module, levels = module_order)

ggplot(Fulltrait_df, aes(x = Module, y = Trait, fill = cor)) +
  geom_tile() +
  # First geom_text for annotations
  geom_text(aes(label = annotation), color = &quot;black&quot;, size = 10) +
  # Second geom_text for correlation values
  geom_text(aes(label = sprintf(&quot;%.2f&quot;, cor)), color = &quot;black&quot;, size = 4, vjust = .8) +
  scale_fill_gradient2(
    low = &quot;blue&quot;, 
    high = &quot;red&quot;, 
    mid = &quot;white&quot;, 
    midpoint = 0, 
    name = &quot;Correlation&quot;,
    limits = c(-1, 1)  # Set the limits for the color scale
  ) +
  labs(x = &quot;Trait&quot;, y = &quot;Module&quot;) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Define%20trat%20and%20get%20eigen%20proteins-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-b---pairwise-differential-abundance-of-proteins"
class="section level1">
<h1>Figure-2-B - Pairwise differential abundance of proteins</h1>
<pre class="r"><code># Get your genes/proteins and modules DF 
Gene_mod_DF &lt;- net$genes_and_modules 

# Module data gets merged with toptable
toptable_summary %&gt;% head()</code></pre>
<pre><code>           logFC  AveExpr         t      P.Value    adj.P.Val        B Protein
Q15061  2.436903 18.60721  26.00939 1.499311e-07 1.499311e-07 7.414993  Q15061
O76021 -2.224860 21.98800 -18.86036 1.191879e-06 1.191879e-06 6.029757  O76021
P43694 -1.821642 18.81578 -15.98230 3.129665e-06 3.129665e-06 5.276514  P43694
P01130 -1.819689 20.01194 -15.77595 3.483331e-06 3.483331e-06 5.186033  P01130
Q96P11 -1.843458 17.28018 -14.33937 5.735316e-06 5.735316e-06 4.764600  Q96P11
Q6EMK4  1.838032 19.30419  14.22737 6.455771e-06 6.455771e-06 4.660093  Q6EMK4
       threshold_P
Q15061        TRUE
O76021        TRUE
P43694        TRUE
P01130        TRUE
Q96P11        TRUE
Q6EMK4        TRUE</code></pre>
<pre class="r"><code>Toptable_Modules &lt;- merge(toptable_summary, 
                          Gene_mod_DF, by.x = &quot;Protein&quot;, by.y = &quot;Genes&quot;)

# Create the volcano plot
ggplot(Toptable_Modules)+
  geom_point(mapping = aes(x = logFC, y = -log10(P.Value), color = threshold_P))+
  xlab(&quot;log2FC&quot;)+
  ylab(&quot;-log10 nominal p-value&quot;)+
  ylim(0, 7)+
  xlim(-3.5, 3.5)+
  theme(legend.position = &quot;none&quot;, 
        plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = rel(1.25)))+
  theme_classic()</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Volcano-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="figure-2-c-module-composition-for-differentilly-abundant-proteins"
class="section level1">
<h1>Figure-2-C Module composition for differentilly abundant
proteins</h1>
<pre class="r"><code># Plotting LogFC between DE and non DE proteins 
Toptable_Modules$DE_or_Not &lt;- Toptable_Modules$P.Value &lt; 0.05


# Calculate percentage of DE genes for each module
DE_Mod_result &lt;- Toptable_Modules %&gt;%
  group_by(Modules) %&gt;%
  summarize(
    total_genes = n(),
    DE_genes = sum(P.Value &lt; 0.05),
    percent_DE = (DE_genes / total_genes) * 100
  )


print(DE_Mod_result)</code></pre>
<pre><code># A tibble: 13 × 4
   Modules      total_genes DE_genes percent_DE
   &lt;chr&gt;              &lt;int&gt;    &lt;int&gt;      &lt;dbl&gt;
 1 blue                  45        1       2.22
 2 brown                 49       10      20.4 
 3 darkgreen            560      319      57.0 
 4 darkred              798       96      12.0 
 5 green                579      315      54.4 
 6 grey                 106        9       8.49
 7 lightgreen           262       14       5.34
 8 lightyellow          437      142      32.5 
 9 magenta              129        0       0   
10 midnightblue         233      122      52.4 
11 royalblue            715       55       7.69
12 salmon               208       84      40.4 
13 yellow                57        0       0   </code></pre>
<pre class="r"><code>DE_Mod_result[DE_Mod_result$Modules %in% c(&quot;green&quot;,&quot;darkgreen&quot;, &quot;midnightblue&quot;, &quot;lightyellow&quot;, &quot;salmon&quot;), ]$percent_DE %&gt;% range()</code></pre>
<pre><code>[1] 32.49428 56.96429</code></pre>
<pre class="r"><code>DE_Mod_result[!(DE_Mod_result$Modules %in% c(&quot;green&quot;,&quot;darkgreen&quot;, &quot;midnightblue&quot;, &quot;lightyellow&quot;, &quot;salmon&quot;)), ]$percent_DE %&gt;% range()</code></pre>
<pre><code>[1]  0.00000 20.40816</code></pre>
<pre class="r"><code># Plot
ggplot(DE_Mod_result, aes(x = Modules, y = percent_DE)) +
  geom_bar(stat = &quot;identity&quot;) +
  labs(
    x = &quot;Module&quot;,
    y = &quot;Percentage of DE genes&quot;,
    title = &quot;Percentage of DE genes in each module&quot;,
    fill = &quot;Module&quot;
  ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/DE%20across%20modules%201-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Compute complementary percentage for non-DE genes
DE_Mod_result$percent_nonDE &lt;- 100 - DE_Mod_result$percent_DE

# Reshape data to long format
long_data &lt;- DE_Mod_result %&gt;% 
  pivot_longer(cols = c(percent_DE, percent_nonDE), 
               names_to = &quot;type&quot;, 
               values_to = &quot;percentage&quot;)

module_order &lt;- c(&quot;green&quot;,&quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;, &quot;lightyellow&quot;, &quot;lightgreen&quot;, &quot;blue&quot;,&quot;magenta&quot;, &quot;darkred&quot;,&quot;brown&quot;, &quot;yellow&quot;,&quot;royalblue&quot;, &quot;grey&quot;)  



# Factor the Module column in Fulltrait_df
long_data$Modules &lt;- factor(long_data$Modules, levels = module_order)

# Plot stacked bar chart
ggplot(long_data, aes(x = Modules, y = percentage, fill = type)) +
  geom_bar(stat = &quot;identity&quot;) +
  labs(
    x = &quot;Module&quot;,
    y = &quot;Percentage&quot;,
    title = &quot;Percentage of DE and non-DE genes in each module&quot;,
    fill = &quot;Type&quot;
  ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_y_continuous(breaks = rev(seq(0, 100, by = 10)))</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/DE%20across%20modules%201-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-d-dox-response-effect-size" class="section level1">
<h1>Figure-2-D DOX-response effect size</h1>
<pre class="r"><code>Toptable_Modules_DOXcorset &lt;- Toptable_Modules[Toptable_Modules$Modules %in% c(&quot;green&quot;, &quot;lightyellow&quot; , &quot;salmon&quot;, &quot;darkgreen&quot; ,&quot;midnightblue&quot;), ]



# Density 
ggplot(Toptable_Modules_DOXcorset, aes(x = logFC, fill = Modules)) +
  geom_density(alpha = 0.9) + 
  labs(title = &quot;Distribution of logFC Values in DOX correlated modules&quot;,
       x = &quot;logFC&quot;,
       y = &quot;Density&quot;) +
  theme_minimal() +
  theme(legend.position = &quot;right&quot;)+
  xlim(-2,2)+
  ylim(0, 4.5)+
  scale_fill_identity(guide = &quot;legend&quot;)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite values (`stat_density()`).</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Ridge%201-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Toptable_Modules_DOXcorset_SAYANversion &lt;- Toptable_Modules_DOXcorset


Toptable_Modules_DOXcorset_SAYANversion$Modules[Toptable_Modules_DOXcorset_SAYANversion$Modules == &quot;lightyellow&quot;] &lt;- &quot;goldenrod&quot; 


# Full graph 
module_order &lt;- c(&quot;green&quot;,&quot;darkgreen&quot;,&quot;midnightblue&quot;, &quot;salmon&quot;,&quot;goldenrod&quot;)

# Factor the Module column in Fulltrait_df
Toptable_Modules_DOXcorset_SAYANversion$Modules &lt;- factor(Toptable_Modules_DOXcorset_SAYANversion$Modules, levels = rev(module_order))



# Density 
ggplot(Toptable_Modules_DOXcorset_SAYANversion, aes(x = logFC, fill = Modules)) +
  geom_density(alpha = 0.9) + 
  labs(title = &quot;Distribution of logFC Values in DOX correlated modules&quot;,
       x = &quot;logFC&quot;,
       y = &quot;Density&quot;) +
  theme_classic() +
  theme(legend.position = &quot;right&quot;)+
  xlim(-2,2)+
  ylim(0, 4)+
  scale_fill_identity(guide = &quot;legend&quot;)</code></pre>
<pre><code>Warning: Removed 10 rows containing non-finite values (`stat_density()`).</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Ridge%201-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Toptable_Modules_NonDOXcorset &lt;- Toptable_Modules[Toptable_Modules$Modules %in% c(&quot;darkred&quot;, &quot;royalblue&quot; ,&quot;lightgreen&quot;, &quot;magenta&quot;, &quot;blue&quot; ,&quot;brown&quot;, &quot;yellow&quot;), ]


# Density
ggplot(Toptable_Modules_NonDOXcorset, aes(x = logFC, fill = Modules)) +
  geom_density(alpha = 0.9 ) + 
  labs(title = &quot;Distribution of logFC Values in modules not correlaated to DOX&quot;,
       x = &quot;logFC&quot;,
       y = &quot;Density&quot;) +
  theme_minimal() +
  theme(legend.position = &quot;right&quot;)+
  xlim(-2,2)+
  ylim(0, 4.5)+
  scale_fill_identity(guide = &quot;legend&quot;)</code></pre>
<pre><code>Warning: Removed 3 rows containing non-finite values (`stat_density()`).</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Ridge%201-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-e-hub-protein-expresssion-examples"
class="section level1">
<h1>Figure-2-E Hub protein expresssion examples</h1>
<pre class="r"><code>boxplot(as.matrix(log2(RUVg_Log2_quantnormalized_all10samples))[&quot;Q96GQ7&quot;,] ~ Meta$Cond, data=Meta, main=&quot;DDX27&quot;,
        ylab=&quot;Log2 Abundance&quot;, xlab=&quot;&quot;, col = c(&quot;green&quot;,&quot;green&quot;), ylim = c(8,9))</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Ridge%202-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(as.matrix(log2(RUVg_Log2_quantnormalized_all10samples))[&quot;Q9BRB3&quot;,] ~ Meta$Cond, data=Meta, main=&quot;PIGQ&quot;,
        ylab=&quot;Log2 Abundance&quot;, xlab=&quot;&quot;, col = c(&quot;yellow&quot;,&quot;yellow&quot;),  ylim = c(8,9))</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/Ridge%202-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="figure-2-f-hub-protein-co-expression-network"
class="section level1">
<h1>Figure-2-F-Hub protein co-expression network</h1>
<pre class="r"><code>#### Version1####
hubs_traitcorr &lt;- hubs[hubs$Module %in% c( &quot;darkgreen&quot;, &quot;green&quot; , &quot;lightyellow&quot;,  &quot;midnightblue&quot;, &quot;salmon&quot;), ]

hubs_traitcorr_toptable &lt;-  merge(hubs_traitcorr, Toptable_Modules, by.x=&quot;Gene&quot; , by.y= &quot;Protein&quot;)


My_edge_list &lt;- BioNERO::get_edge_list(net)

# Get a data frame of DOX corr. hubs connecting all to all 
My_edge_list_hubs &lt;- My_edge_list %&gt;%
  filter(Gene1 %in% hubs_traitcorr$Gene &amp; Gene2 %in% hubs_traitcorr$Gene)
# looks like it is too big for plotting try something else 



# Create a helper dataframe with Gene and Module from hubs_traitcorr
helper_df &lt;- hubs_traitcorr %&gt;% dplyr::select(Gene, Module)

# Now, subset My_edge_list based on the conditions
My_edge_list_hubs_same_module &lt;- My_edge_list_hubs %&gt;%
  dplyr::filter(
    Gene1 %in% helper_df$Gene &amp; 
    Gene2 %in% helper_df$Gene
  ) %&gt;%
  dplyr::filter(
    helper_df$Module[match(Gene1, helper_df$Gene)] == 
    helper_df$Module[match(Gene2, helper_df$Gene)]
  )

My_edge_list_hubs_same_module %&gt;% head()</code></pre>
<pre><code>   Gene1  Gene2    Weight
1 O00567 O14807 0.8746559
2 O00571 O15121 0.9570017
3 O00571 O15371 0.9484436
4 O15121 O15371 0.9756670
5 O00571 O15372 0.9640689
6 O15121 O15372 0.9737202</code></pre>
<pre class="r"><code>My_edge_list_hubs_same_module_filt &lt;- My_edge_list_hubs_same_module[My_edge_list_hubs_same_module$Weight &gt; 0.9, ] 



#### Version 2 fully connected #### 
# Create an igraph graph object

graph_network &lt;- graph_from_data_frame(d = My_edge_list_hubs[My_edge_list_hubs$Weight &gt; 0.9, ], directed = FALSE)

# Prepare the node attribute data frame with gene and module information
node_attributes &lt;- dplyr::select(hubs_traitcorr, Gene, Module) %&gt;% unique()

# Add the module information to the graph
V(graph_network)$Module &lt;- node_attributes$Module[match(names(V(graph_network)), node_attributes$Gene)]


ggraph(graph_network, layout = &quot;stress&quot;) +
  geom_edge_link(aes(edge_color = Weight), show.legend = TRUE) +
  geom_node_point(aes(color = Module), stroke = 0.5) +  # Color nodes by module
  geom_node_text(aes(label = name), repel = TRUE, size = 2) +
  scale_edge_color_continuous(low = &quot;lightgrey&quot;, high = &quot;black&quot;, limits = c(0.9, 1)) +  # Scale edge color by weight
  scale_color_manual(values = unique(hubs_traitcorr$Module)) +  # Set colors for each module
  theme_graph() +
  labs(color = &quot;Module&quot;)</code></pre>
<pre><code>Warning: Using the `size` aesthetic in this geom was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` in the `default_aes` field and elsewhere instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: ggrepel: 43 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/2F-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggraph(graph_network, layout = &quot;stress&quot;) +
  geom_edge_link(aes(edge_color = Weight, alpha = Weight), show.legend = TRUE) +  # Edge color and alpha by Weight
  geom_node_point(aes(color = Module), size = 2) +  # Color nodes by module, adjust node size
  geom_node_text(aes(label = name), repel = TRUE, size = 2) +  # Adjust text size
  scale_edge_color_continuous(low = &quot;white&quot;, high = &quot;black&quot;, limits = c(0.9, 1)) +  # Scale edge color by weight
  scale_color_manual(values = unique(hubs_traitcorr$Module)) +  # Set colors for each module
  theme_graph() +
  labs(color = &quot;Module&quot;)</code></pre>
<pre><code>Warning: ggrepel: 43 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/Johnson_DOX_24_2.Rmd/2F-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggraph_2.1.0                           
 [2] igraph_1.5.1                           
 [3] ReactomePA_1.40.0                      
 [4] impute_1.70.0                          
 [5] WGCNA_1.72-1                           
 [6] fastcluster_1.2.3                      
 [7] dynamicTreeCut_1.63-1                  
 [8] BioNERO_1.4.2                          
 [9] reshape2_1.4.4                         
[10] ggridges_0.5.4                         
[11] biomaRt_2.52.0                         
[12] ggvenn_0.1.10                          
[13] UpSetR_1.4.0                           
[14] DOSE_3.22.1                            
[15] variancePartition_1.26.0               
[16] clusterProfiler_4.4.4                  
[17] pheatmap_1.0.12                        
[18] qvalue_2.28.0                          
[19] Homo.sapiens_1.3.1                     
[20] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[21] org.Hs.eg.db_3.15.0                    
[22] GO.db_3.15.0                           
[23] OrganismDbi_1.38.1                     
[24] GenomicFeatures_1.48.4                 
[25] AnnotationDbi_1.58.0                   
[26] cluster_2.1.4                          
[27] ggfortify_0.4.16                       
[28] lubridate_1.9.2                        
[29] forcats_1.0.0                          
[30] stringr_1.5.0                          
[31] dplyr_1.1.2                            
[32] purrr_1.0.2                            
[33] readr_2.1.4                            
[34] tidyr_1.3.0                            
[35] tibble_3.2.1                           
[36] ggplot2_3.4.3                          
[37] tidyverse_2.0.0                        
[38] RColorBrewer_1.1-3                     
[39] RUVSeq_1.30.0                          
[40] edgeR_3.38.4                           
[41] limma_3.52.4                           
[42] EDASeq_2.30.0                          
[43] ShortRead_1.54.0                       
[44] GenomicAlignments_1.32.1               
[45] SummarizedExperiment_1.26.1            
[46] MatrixGenerics_1.8.1                   
[47] matrixStats_1.0.0                      
[48] Rsamtools_2.12.0                       
[49] GenomicRanges_1.48.0                   
[50] Biostrings_2.64.1                      
[51] GenomeInfoDb_1.32.4                    
[52] XVector_0.36.0                         
[53] IRanges_2.30.1                         
[54] S4Vectors_0.34.0                       
[55] BiocParallel_1.30.4                    
[56] Biobase_2.56.0                         
[57] BiocGenerics_0.42.0                    
[58] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] enrichplot_1.16.2      xml2_1.3.5             httpuv_1.6.11         
 [28] viridis_0.6.4          xfun_0.40              hms_1.1.3             
 [31] jquerylib_0.1.4        evaluate_0.21          promises_1.2.1        
 [34] fansi_1.0.4            restfulr_0.0.15        progress_1.2.2        
 [37] caTools_1.18.2         dbplyr_2.3.3           htmlwidgets_1.6.2     
 [40] DBI_1.1.3              ggnewscale_0.4.9       backports_1.4.1       
 [43] annotate_1.74.0        aod_1.3.2              deldir_1.0-9          
 [46] vctrs_0.6.3            abind_1.4-5            cachem_1.0.8          
 [49] withr_2.5.0            ggforce_0.4.1          checkmate_2.2.0       
 [52] treeio_1.20.2          prettyunits_1.1.1      ape_5.7-1             
 [55] lazyeval_0.2.2         crayon_1.5.2           genefilter_1.78.0     
 [58] labeling_0.4.2         pkgconfig_2.0.3        tweenr_2.0.2          
 [61] nlme_3.1-163           nnet_7.3-19            rlang_1.1.1           
 [64] lifecycle_1.0.3        downloader_0.4         filelock_1.0.2        
 [67] BiocFileCache_2.4.0    rprojroot_2.0.3        polyclip_1.10-4       
 [70] graph_1.74.0           Matrix_1.5-4.1         aplot_0.2.0           
 [73] NetRep_1.2.7           boot_1.3-28.1          base64enc_0.1-3       
 [76] GlobalOptions_0.1.2    whisker_0.4.1          processx_3.8.2        
 [79] png_0.1-8              viridisLite_0.4.2      rjson_0.2.21          
 [82] bitops_1.0-7           getPass_0.2-2          R.oo_1.25.0           
 [85] ggnetwork_0.5.12       KernSmooth_2.23-22     blob_1.2.4            
 [88] shape_1.4.6            jpeg_0.1-10            gridGraphics_0.5-1    
 [91] reactome.db_1.81.0     scales_1.2.1           graphite_1.42.0       
 [94] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.8            
 [97] gplots_3.1.3           zlibbioc_1.42.0        compiler_4.2.0        
[100] scatterpie_0.2.1       BiocIO_1.6.0           clue_0.3-64           
[103] intergraph_2.0-3       lme4_1.1-34            cli_3.6.1             
[106] patchwork_1.1.3        ps_1.7.5               htmlTable_2.4.1       
[109] Formula_1.2-5          mgcv_1.9-0             MASS_7.3-60           
[112] tidyselect_1.2.0       stringi_1.7.12         highr_0.10            
[115] yaml_2.3.7             GOSemSim_2.22.0        locfit_1.5-9.8        
[118] latticeExtra_0.6-30    ggrepel_0.9.3          sass_0.4.7            
[121] fastmatch_1.1-4        tools_4.2.0            timechange_0.2.0      
[124] parallel_4.2.0         circlize_0.4.15        rstudioapi_0.15.0     
[127] foreign_0.8-84         foreach_1.5.2          git2r_0.32.0          
[130] gridExtra_2.3          farver_2.1.1           digest_0.6.33         
[133] BiocManager_1.30.22    networkD3_0.4          Rcpp_1.0.11           
[136] broom_1.0.5            later_1.3.1            httr_1.4.7            
[139] ComplexHeatmap_2.12.1  GENIE3_1.18.0          Rdpack_2.5            
[142] colorspace_2.1-0       XML_3.99-0.14          fs_1.6.3              
[145] splines_4.2.0          statmod_1.5.0          yulab.utils_0.0.8     
[148] RBGL_1.72.0            tidytree_0.4.5         graphlayouts_1.0.0    
[151] ggplotify_0.1.2        xtable_1.8-4           jsonlite_1.8.7        
[154] nloptr_2.0.3           ggtree_3.4.4           tidygraph_1.2.3       
[157] ggfun_0.1.2            R6_2.5.1               Hmisc_5.1-0           
[160] pillar_1.9.0           htmltools_0.5.6        glue_1.6.2            
[163] fastmap_1.1.1          minqa_1.2.5            codetools_0.2-19      
[166] fgsea_1.22.0           utf8_1.2.3             sva_3.44.0            
[169] lattice_0.21-8         bslib_0.5.1            network_1.18.1        
[172] pbkrtest_0.5.2         curl_5.0.2             gtools_3.9.4          
[175] interp_1.1-4           survival_3.5-7         statnet.common_4.9.0  
[178] rmarkdown_2.24         munsell_0.5.0          GetoptLong_1.0.5      
[181] DO.db_2.9              GenomeInfoDbData_1.2.8 iterators_1.0.14      
[184] gtable_0.3.4           rbibutils_2.2.15      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
