<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Omar Johnson" />

<meta name="date" content="2024-07-21" />

<title>Johnson_DOX_24_5</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Johnson_DOX_24</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Johnson_DOX_24_5</h1>
<h4 class="author">Omar Johnson</h4>
<h4 class="date">2024-07-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DOX_24_Github/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240723code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240723)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240723code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240723)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong88c6686">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 88c6686
</a>
</p>
</div>
<div id="strongRepositoryversionstrong88c6686"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 88c6686.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/Johnson_DOX_24_5.html
    Ignored:    analysis/Johnson_DOX_24_6.html
    Ignored:    analysis/Johnson_DOX_24_7.html
    Ignored:    analysis/Johnson_DOX_24_8.html
    Ignored:    analysis/Johnson_DOX_24_RUV_Limma.html

Untracked files:
    Untracked:  Fig_2.Rmd
    Untracked:  Fig_2.html
    Untracked:  analysis/Johnson_DOX_24_RUV_Limma.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Johnson_DOX_24_5.Rmd</code>) and
HTML (<code>docs/Johnson_DOX_24_5.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
88c6686
</td>
<td>
Omar-Johnson
</td>
<td>
2024-07-23
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load Libraries</h1>
</div>
<div id="read-in-data" class="section level1">
<h1>Read in Data</h1>
</div>
<div id="functions" class="section level1">
<h1>Functions</h1>
<pre class="r"><code>perform_fisher_test_FP &lt;- function(vec1, vec2, vec1_name, vec2_name, plot = FALSE) {
  # Create labeled factors for vec1 and vec2
  vec1_label &lt;- factor(vec1, labels = c(paste0(&quot;Not&quot;, vec1_name), paste0(&quot;Is&quot;, vec1_name)))
  vec2_label &lt;- factor(vec2, labels = c(paste0(&quot;Not&quot;, vec2_name), paste0(&quot;Is&quot;, vec2_name)))

  # Create contingency table with labeled factors
  table &lt;- table(vec1_label, vec2_label)

  # Perform Fisher&#39;s exact test
  test_result &lt;- fisher.test(table)
  p_value &lt;- test_result$p.value
OR &lt;- test_result$estimate
CI &lt;- test_result$conf.int

  # Prepare result
  result &lt;- list(
    ContingencyTable = table,
    PValue = p_value, 
    Odds_ratio = test_result$estimate,
    Confidence_Interval = test_result$conf.int
  )

  # Generate plot if required
  if (plot) {
    # Convert table to data frame for ggplot
    table_df &lt;- as.data.frame(as.table(table))
    colnames(table_df) &lt;- c(&quot;vec1_label&quot;, &quot;vec2_label&quot;, &quot;Freq&quot;)

    # Calculate totals for each vec1_label
    totals &lt;- aggregate(Freq ~ vec1_label, data = table_df, sum)

    # Merge totals with table_df and calculate percentages
    table_df &lt;- merge(table_df, totals, by = &quot;vec1_label&quot;, all.x = TRUE)
    table_df$Percentage &lt;- with(table_df, Freq.x / Freq.y * 100)
    table_df$Group &lt;- table_df$vec2_label

    # Stacked bar chart
    p &lt;- ggplot(table_df, aes(x = vec1_label, y = Percentage, fill = Group)) +
      geom_bar(stat = &quot;identity&quot;, position = &quot;stack&quot;) +  # Adjust position to &quot;stack&quot;
      facet_wrap(~ vec1_label) +
      theme_minimal() +
      labs(x = vec1_name, y = &quot;Percentage&quot;, fill = vec2_name, title = paste(&quot;&quot;)) +
      theme(axis.text.x = element_text(angle = 45, hjust = 1))

    result$Plot &lt;- p
  }

  return(result)
}


group_by_deciles &lt;- function(x) {
  deciles &lt;- cut(x, 
                 breaks = quantile(x, probs = seq(0, 1, by = 0.1), na.rm = TRUE), 
                 include.lowest = TRUE, 
                 labels = paste0(&quot;D&quot;, 1:10))
  return(deciles)
}</code></pre>
</div>
<div id="fig-4b-pqtl-enrichment-test" class="section level1">
<h1>Fig-4B pQTL enrichment test</h1>
<pre class="r"><code>#### Forest plots #### 

pQTL_data &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/pQTL_Genetic_reg_plasmaproteome.csv&quot;, header = TRUE)

# Get the data frame with expressed proteins in the set 
pQTL_data_exp &lt;- pQTL_data[pQTL_data$Target.UniProt %in% Toptable_Modules$Protein, ]

# Get All proteins mapped to at least one cis and one trans  
pQTL_data_Trans &lt;- pQTL_data_exp[pQTL_data_exp$cis.trans == &quot;trans&quot;, ]
pQTL_data_Cis &lt;- pQTL_data_exp[pQTL_data_exp$cis.trans == &quot;cis&quot;, ]

# Count them 
All_pQTL &lt;- pQTL_data$Target.UniProt %&gt;% unique()
Trans_pQTL &lt;- pQTL_data_Trans$Target.UniProt %&gt;% unique()
Cis_pQTL &lt;- pQTL_data_Cis$Target.UniProt %&gt;% unique()


# Get Sets 
pQTL_data_summary_merged &lt;- Toptable_Modules


# Append data frame to contain pQTL info 
pQTL_data_summary_merged &lt;- pQTL_data_summary_merged %&gt;%
  mutate(All_pQTL = if_else(pQTL_data_summary_merged$Protein %in% All_pQTL, 1, 0))

pQTL_data_summary_merged &lt;- pQTL_data_summary_merged %&gt;%
  mutate(Trans_pQTL = if_else(pQTL_data_summary_merged$Protein %in% Trans_pQTL, 1, 0))

pQTL_data_summary_merged &lt;- pQTL_data_summary_merged %&gt;%
  mutate(Cis_pQTL = if_else(pQTL_data_summary_merged$Protein %in% Cis_pQTL, 1, 0))




# Annotate if protein is hub
pQTL_data_summary_merged &lt;- pQTL_data_summary_merged %&gt;%
  mutate(Is_hub = if_else(pQTL_data_summary_merged$Protein %in% hubs$Gene, 1, 0))

# Annotate if protein is DA
pQTL_data_summary_merged &lt;- pQTL_data_summary_merged %&gt;%
  mutate(Is_DA = if_else(P.Value &lt; 0.05, 1, 0))

# Annotate if protein is DOX corr. 
pQTL_data_summary_merged &lt;- pQTL_data_summary_merged %&gt;%
  mutate(Is_DOXcorr = if_else( Modules %in% c(&quot;green&quot;,&quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;), 1, 0))


# Get data frame of just the hub proteins 
pQTL_data_summary_merged_hub &lt;- pQTL_data_summary_merged[pQTL_data_summary_merged$Protein %in% hubs$Gene, ]


# Cis
FP_Cis_DOXcor &lt;- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged$Is_DOXcorr, vec2 = pQTL_data_summary_merged$Cis_pQTL, vec1_name = &quot;&quot;, vec2_name = &quot;&quot;,  plot = FALSE)


FP_Cis_Hub &lt;- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged$Is_hub, vec2 = pQTL_data_summary_merged$Cis_pQTL, vec1_name = &quot;&quot;, vec2_name = &quot;&quot;,  plot = FALSE)



FP_Cis_DOXcorhub &lt;- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged_hub$Is_DOXcorr, vec2 = pQTL_data_summary_merged_hub$Cis_pQTL, vec1_name = &quot;&quot;, vec2_name = &quot;&quot;,  plot = FALSE)




# Trans
FP_Trans_DOXcor &lt;- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged$Is_DOXcorr, vec2 = pQTL_data_summary_merged$Trans_pQTL, vec1_name = &quot;&quot;, vec2_name = &quot;&quot;,  plot = FALSE)


FP_Trans_Hub &lt;- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged$Is_hub, vec2 = pQTL_data_summary_merged$Trans_pQTL, vec1_name = &quot;&quot;, vec2_name = &quot;&quot;,  plot = FALSE)

FP_Trans_DOXcorhub &lt;- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged_hub$Is_DOXcorr, vec2 = pQTL_data_summary_merged_hub$Trans_pQTL, vec1_name = &quot;&quot;, vec2_name = &quot;&quot;,  plot = FALSE)


FP_List &lt;- list(FP_Cis_DOXcor, FP_Cis_Hub, FP_Cis_DOXcorhub, FP_Trans_DOXcor, FP_Trans_Hub, FP_Trans_DOXcorhub)

FP_DF &lt;- data.frame(
  Odds_ratio = numeric(length(FP_List)),
  Lower_CI = numeric(length(FP_List)),
  Upper_CI = numeric(length(FP_List)), 
  Pval = numeric(length(FP_List))
)

for (i in 1:length(FP_List)) {
  FP_DF$Odds_ratio[i] &lt;- FP_List[[i]]$Odds_ratio
  FP_DF$Lower_CI[i] &lt;- FP_List[[i]]$Confidence_Interval[1]
  FP_DF$Upper_CI[i] &lt;- FP_List[[i]]$Confidence_Interval[2]
  FP_DF$Pval[i] &lt;- FP_List[[i]]$PValue
}

FP_DF</code></pre>
<pre><code>  Odds_ratio  Lower_CI  Upper_CI         Pval
1  0.7002624 0.5768312 0.8488705 2.156762e-04
2  1.0782950 0.7770841 1.4725084 6.304725e-01
3  0.2255299 0.1010994 0.4648597 5.834571e-06
4  0.8169640 0.6879394 0.9695398 1.910247e-02
5  1.0402191 0.7737982 1.3815625 7.731600e-01
6  0.3507411 0.1857923 0.6404221 2.373421e-04</code></pre>
<pre class="r"><code># Add row names for the labels in the forest plot
FP_DF$Label &lt;- c(&quot;FP_Cis_DOXcor&quot;, &quot;FP_Cis_Hub&quot;, &quot;FP_Cis_DOXcorhub&quot;, &quot;FP_Trans_DOXcor&quot;, &quot;FP_Trans_Hub&quot;, &quot;FP_Trans_DOXcorhub&quot;)



FP_DF$Label &lt;- factor(FP_DF$Label, levels = rev(c(&quot;FP_Cis_Hub&quot;, &quot;FP_Trans_Hub&quot;,&quot;FP_Cis_DOXcor&quot;, &quot;FP_Trans_DOXcor&quot;, &quot;FP_Cis_DOXcorhub&quot;, &quot;FP_Trans_DOXcorhub&quot;)))

ggplot(FP_DF, aes(x = Label, y = Odds_ratio, ymin = Lower_CI, ymax = Upper_CI)) +
  geom_pointrange() +
  geom_hline(yintercept = 1, linetype = &quot;dashed&quot;, color = &quot;red&quot;) +
  coord_flip() +
  labs(
    title = &quot;Forest Plot of Odds Ratios&quot;,
    x = &quot;&quot;,
    y = &quot;Odds Ratio (95% CI)&quot;
  ) +
  theme_minimal()</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/pqtl%20enrichment%20across%20network%20proteins-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FP_DF</code></pre>
<pre><code>  Odds_ratio  Lower_CI  Upper_CI         Pval              Label
1  0.7002624 0.5768312 0.8488705 2.156762e-04      FP_Cis_DOXcor
2  1.0782950 0.7770841 1.4725084 6.304725e-01         FP_Cis_Hub
3  0.2255299 0.1010994 0.4648597 5.834571e-06   FP_Cis_DOXcorhub
4  0.8169640 0.6879394 0.9695398 1.910247e-02    FP_Trans_DOXcor
5  1.0402191 0.7737982 1.3815625 7.731600e-01       FP_Trans_Hub
6  0.3507411 0.1857923 0.6404221 2.373421e-04 FP_Trans_DOXcorhub</code></pre>
<pre class="r"><code># Add a new column to indicate &quot;cis&quot; or &quot;trans&quot;
FP_DF$Type &lt;- ifelse(grepl(&quot;Cis&quot;, FP_DF$Label), &quot;Cis-pQTL&quot;, &quot;Trans-pQTL&quot;)

# Reorder the levels
FP_DF$Label &lt;- factor(FP_DF$Label, levels = rev(c(&quot;FP_Cis_Hub&quot;, &quot;FP_Trans_Hub&quot;, &quot;FP_Cis_DOXcor&quot;, &quot;FP_Trans_DOXcor&quot;, &quot;FP_Cis_DOXcorhub&quot;, &quot;FP_Trans_DOXcorhub&quot;)))

# Plot with color based on &quot;Type&quot;
ggplot(FP_DF, aes(x = Label, y = Odds_ratio, ymin = Lower_CI, ymax = Upper_CI, color = Type)) +
  geom_pointrange() +
  geom_hline(yintercept = 1, linetype = &quot;dashed&quot;, color = &quot;red&quot;) +
  coord_flip() +
  labs(
    title = &quot;pQTL enrichment among network proteins&quot;,
    x = &quot;&quot;,
    y = &quot;Odds of being pQTL (95% CI)&quot;
  ) +
  theme_classic() +
  scale_color_manual(values = c(&quot;Cis-pQTL&quot; = &quot;dodgerblue4&quot;, &quot;Trans-pQTL&quot; = &quot;orange3&quot;))</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/pqtl%20enrichment%20across%20network%20proteins-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fig-4c-cistran-pqtl-snp-beta" class="section level1">
<h1>Fig-4C cis/tran pQTL SNP beta</h1>
<pre class="r"><code>#### Cis or trans SNP-Effect size #### 

# Read in data from UK biobank paper: https://doi.org/10.1038/s41586-023-06592-6 
pQTL_data &lt;- read.csv(file = &quot;/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/GWAS/pQTL_Genetic_reg_plasmaproteome.csv&quot;, header = TRUE)

# Get the data frame with expressed proteins in the set 
pQTL_data_exp &lt;- pQTL_data[pQTL_data$Target.UniProt %in% Toptable_Modules$Protein, ]

# Get All proteins mapped to at least one cis and one trans  
pQTL_data_Trans &lt;- pQTL_data_exp[pQTL_data_exp$cis.trans == &quot;trans&quot;, ]
pQTL_data_Cis &lt;- pQTL_data_exp[pQTL_data_exp$cis.trans == &quot;cis&quot;, ]

# Summarize All pQTL effect sizes 
pQTL_data_summary &lt;- pQTL_data_exp %&gt;%
  group_by(Target.UniProt) %&gt;%
  arrange(BETA) %&gt;%
  summarise(
    Mean_BETA = mean(abs(BETA), na.rm = TRUE),
    Median_BETA = median(abs(BETA), na.rm = TRUE),
    Max_BETA = max(abs(BETA), na.rm = TRUE)
  )
pQTL_data_summary$Max_BETA %&gt;% hist()</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/cis/tran%20pQTL%20SNP%20beta-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Summarize Trans pQTL effect sizes
Trans_pQTL_data_summary &lt;- pQTL_data_Trans %&gt;%
  group_by(Target.UniProt) %&gt;%
  arrange(BETA) %&gt;%
  summarise(
    Mean_BETA = mean(abs(BETA), na.rm = TRUE),
    Median_BETA = median(abs(BETA), na.rm = TRUE),
    Max_BETA = max(abs(BETA), na.rm = TRUE)
  )
Trans_pQTL_data_summary$Max_BETA %&gt;% hist()</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/cis/tran%20pQTL%20SNP%20beta-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Trans_pQTL_data_summary$Max_BETA %&gt;% median()</code></pre>
<pre><code>[1] 0.0985</code></pre>
<pre class="r"><code># Summarize Cis pQTL effect sizes
Cis_pQTL_data_summary &lt;- pQTL_data_Cis %&gt;%
  group_by(Target.UniProt) %&gt;%
  arrange(BETA) %&gt;%
  summarise(
    Mean_BETA = mean(abs(BETA), na.rm = TRUE),
    Median_BETA = median(abs(BETA), na.rm = TRUE),
    Max_BETA = max(abs(BETA), na.rm = TRUE)
  )
Cis_pQTL_data_summary$Max_BETA %&gt;% hist()</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/cis/tran%20pQTL%20SNP%20beta-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Cis_pQTL_data_summary$Max_BETA %&gt;% median()</code></pre>
<pre><code>[1] 0.2295</code></pre>
<pre class="r"><code># Merge pQTL SNP Data with Toptable_Modules- Cis or Trans
pQTL_data_merged &lt;- merge(Toptable_Modules, pQTL_data_summary, by.x =&quot;Protein&quot; , by.y =&quot;Target.UniProt&quot; )

Cis_pQTL_data_merged &lt;- merge(Toptable_Modules, Cis_pQTL_data_summary, by.x =&quot;Protein&quot; , by.y =&quot;Target.UniProt&quot; )

Trans_pQTL_data_merged &lt;- merge(Toptable_Modules, Trans_pQTL_data_summary, by.x =&quot;Protein&quot; , by.y =&quot;Target.UniProt&quot; )






# Hub vs not hub 
# Network differences in effect size 
All_pQTL_DAPs &lt;- pQTL_data_merged[pQTL_data_merged$Protein %in% hubs$Gene,  ]
All_pQTL_NonDAPs &lt;- pQTL_data_merged[!pQTL_data_merged$Protein %in% hubs$Gene,  ]

Cis_pQTL_DAPs &lt;- Cis_pQTL_data_merged[Cis_pQTL_data_merged$Protein %in% hubs$Gene,  ]
Cis_pQTL_NonDAPs &lt;- Cis_pQTL_data_merged[!Cis_pQTL_data_merged$Protein %in% hubs$Gene,  ]

Trans_pQTL_DAPs &lt;- Trans_pQTL_data_merged[Trans_pQTL_data_merged$Protein %in% hubs$Gene,  ]
Trans_pQTL_NonDAPs &lt;- Trans_pQTL_data_merged[!Trans_pQTL_data_merged$Protein %in% hubs$Gene,  ]


# Test 
# 1. 
wilcox.test(All_pQTL_DAPs$Max_BETA , All_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  All_pQTL_DAPs$Max_BETA and All_pQTL_NonDAPs$Max_BETA
W = 23719, p-value = 0.77
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># 2. 
wilcox.test(Cis_pQTL_DAPs$Max_BETA , Cis_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Cis_pQTL_DAPs$Max_BETA and Cis_pQTL_NonDAPs$Max_BETA
W = 12867, p-value = 0.3136
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># 3. 
wilcox.test(Trans_pQTL_DAPs$Max_BETA , Trans_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Trans_pQTL_DAPs$Max_BETA and Trans_pQTL_NonDAPs$Max_BETA
W = 19615, p-value = 0.744
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># DOX corr. vs not DOX corr.  
# Network differences in effect size 

All_pQTL_DAPs &lt;- pQTL_data_merged[pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;),  ]
All_pQTL_NonDAPs &lt;- pQTL_data_merged[!pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;),  ]


Cis_pQTL_DAPs &lt;- Cis_pQTL_data_merged[Cis_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;),  ]
Cis_pQTL_NonDAPs &lt;- Cis_pQTL_data_merged[!Cis_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;),  ]



Trans_pQTL_DAPs &lt;- Trans_pQTL_data_merged[Trans_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;),  ]
Trans_pQTL_NonDAPs &lt;- Trans_pQTL_data_merged[!Trans_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;),  ]





# Test 
# 1. 
wilcox.test(All_pQTL_DAPs$Max_BETA , All_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  All_pQTL_DAPs$Max_BETA and All_pQTL_NonDAPs$Max_BETA
W = 54728, p-value = 0.02904
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># 2. 
wilcox.test(Cis_pQTL_DAPs$Max_BETA , Cis_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Cis_pQTL_DAPs$Max_BETA and Cis_pQTL_NonDAPs$Max_BETA
W = 29894, p-value = 0.456
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># 3. 
wilcox.test(Trans_pQTL_DAPs$Max_BETA , Trans_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Trans_pQTL_DAPs$Max_BETA and Trans_pQTL_NonDAPs$Max_BETA
W = 48899, p-value = 0.1127
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># Now showing only the ones that you want
# Create a data frame
pLI_Hubs_DF_Boxplot &lt;- data.frame(
  # 1. Generate values to compare 
  values = c( 
             Cis_pQTL_DAPs$Max_BETA, 
             Cis_pQTL_NonDAPs$Max_BETA, 
             Trans_pQTL_DAPs$Max_BETA, 
             Trans_pQTL_NonDAPs$Max_BETA),
  
  # 2. Factor values to compare 
  group = factor(
                c(rep(&quot;Cis_DOXcor.&quot;, length(Cis_pQTL_DAPs$Max_BETA)),
                c(rep(&quot;Cis_Not_DOXcor.&quot;, length(Cis_pQTL_NonDAPs$Max_BETA)),
                c(rep(&quot;Trans_DOXcor.&quot;, length(Trans_pQTL_DAPs$Max_BETA)),
                c(rep(&quot;Trans_Not_DOXcor.&quot;,length(Trans_pQTL_NonDAPs$Max_BETA))
                  
                             
                             ))))))




# Create boxplot
ggplot(pLI_Hubs_DF_Boxplot, aes(x = group, y = values)) +
  geom_boxplot() +
  labs(x = &quot;Group&quot;, y = &quot;pQTL-Max-BETA&quot;) +
  ggtitle(&quot;pQTL max SNP effect size: DOXcor. vs Not DOXcor.&quot;)+
  theme_classic()+
  coord_cartesian(ylim = c(0, 1.0))+
  theme(axis.text.x = element_text(color = &quot;grey20&quot;, size = 10, angle = 90, hjust = .5, vjust = .5, face = &quot;plain&quot;))</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/cis/tran%20pQTL%20SNP%20beta-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># DOX corr. vs not DOX corr.  hubs
All_pQTL_DAPs &lt;- pQTL_data_merged[(pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;)) &amp; (pQTL_data_merged$Protein %in% hubs$Gene),  ]

All_pQTL_NonDAPs &lt;- pQTL_data_merged[(!pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;)) &amp; (pQTL_data_merged$Protein %in% hubs$Gene) ,  ]


Cis_pQTL_DAPs &lt;- Cis_pQTL_data_merged[(Cis_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;)) &amp; (Cis_pQTL_data_merged$Protein %in% hubs$Gene) ,  ]

Cis_pQTL_NonDAPs &lt;- Cis_pQTL_data_merged[(!Cis_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;)) &amp; (Cis_pQTL_data_merged$Protein %in% hubs$Gene) ,  ]


Trans_pQTL_DAPs &lt;- Trans_pQTL_data_merged[(Trans_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;)) &amp; (Trans_pQTL_data_merged$Protein %in% hubs$Gene) ,  ]

Trans_pQTL_NonDAPs &lt;- Trans_pQTL_data_merged[(!Trans_pQTL_data_merged$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;)) &amp; (Trans_pQTL_data_merged$Protein %in% hubs$Gene),  ]



# Test 
# 1. 
wilcox.test(All_pQTL_DAPs$Max_BETA , All_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  All_pQTL_DAPs$Max_BETA and All_pQTL_NonDAPs$Max_BETA
W = 286.5, p-value = 0.001232
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># 2. 
wilcox.test(Cis_pQTL_DAPs$Max_BETA , Cis_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>Warning in wilcox.test.default(Cis_pQTL_DAPs$Max_BETA,
Cis_pQTL_NonDAPs$Max_BETA): cannot compute exact p-value with ties</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Cis_pQTL_DAPs$Max_BETA and Cis_pQTL_NonDAPs$Max_BETA
W = 106.5, p-value = 0.007926
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># 3. 
wilcox.test(Trans_pQTL_DAPs$Max_BETA , Trans_pQTL_NonDAPs$Max_BETA)</code></pre>
<pre><code>Warning in wilcox.test.default(Trans_pQTL_DAPs$Max_BETA,
Trans_pQTL_NonDAPs$Max_BETA): cannot compute exact p-value with ties</code></pre>
<pre><code>
    Wilcoxon rank sum test with continuity correction

data:  Trans_pQTL_DAPs$Max_BETA and Trans_pQTL_NonDAPs$Max_BETA
W = 277.5, p-value = 0.02179
alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code># Now showing only the ones that you want: 
# Create a data frame
pLI_Hubs_DF_Boxplot &lt;- data.frame(
  # 1. Generate values to compare 
  values = c( 
             Cis_pQTL_DAPs$Max_BETA, 
             Cis_pQTL_NonDAPs$Max_BETA, 
             Trans_pQTL_DAPs$Max_BETA, 
             Trans_pQTL_NonDAPs$Max_BETA
             ),
  
  # 2. Factor values to compare 
  group = factor(
                c(rep(&quot;Cis_DOXcor.Hub&quot;, length(Cis_pQTL_DAPs$Max_BETA)),
                c(rep(&quot;Cis_Not_DOXcor. Hub&quot;, length(Cis_pQTL_NonDAPs$Max_BETA)),
                c(rep(&quot;Trans_DOXcor. Hub&quot;, length(Trans_pQTL_DAPs$Max_BETA)),
                c(rep(&quot;Trans_Not_DOXcor. Hub&quot;,length(Trans_pQTL_NonDAPs$Max_BETA))
                             ))))))



# Create boxplot
ggplot(pLI_Hubs_DF_Boxplot, aes(x = group, y = values)) +
  geom_boxplot() +
  labs(x = &quot;&quot;, y = &quot;Max-BETA&quot;) +
  ggtitle(&quot;pQTL max SNP effects Hub-DOXcor. vs Hub-Not DOXcor.&quot;)+
  theme_classic()+
  coord_cartesian(ylim = c(0, 1.0))+
  theme(axis.text.x = element_text(color = &quot;grey20&quot;, size = 10, angle = 90, hjust = .5, vjust = .5, face = &quot;plain&quot;))</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/cis/tran%20pQTL%20SNP%20beta-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fig-4d-pqtls-across-connectivities" class="section level1">
<h1>Fig-4D pQTLs across connectivities</h1>
<pre class="r"><code>#### pQTL connectivity #### 
# Function to generate decile labels

pQTL_pro_unique &lt;- pQTL_data$Target.UniProt %&gt;% unique()
kIN_DF_TT &lt;- Toptable_Modules
kIN_DF_TT2 &lt;- kIN_DF_TT

# Generate deciles
kIN_DF_TT2$Decile &lt;- group_by_deciles(kIN_DF_TT2$Norm_kOut)
pQTL_pro_unique_exp &lt;- intersect(pQTL_pro_unique, Toptable_Modules$Protein)




# Check if a protein is a cis or trans pQTL protein
kIN_DF_TT2$is_cis_pQTL &lt;- kIN_DF_TT2$Protein %in% Cis_pQTL_data_summary$Target.UniProt
kIN_DF_TT2$is_trans_pQTL &lt;- kIN_DF_TT2$Protein %in% Trans_pQTL_data_summary$Target.UniProt

# Calculate the total number of proteins in each decile
total_proteins_per_decile &lt;- kIN_DF_TT2 %&gt;%
  group_by(Decile) %&gt;%
  summarise(total_proteins = n())
total_proteins_per_decile</code></pre>
<pre><code># A tibble: 10 × 2
   Decile total_proteins
   &lt;fct&gt;           &lt;int&gt;
 1 D1                418
 2 D2                418
 3 D3                418
 4 D4                417
 5 D5                418
 6 D6                418
 7 D7                417
 8 D8                418
 9 D9                418
10 D10               418</code></pre>
<pre class="r"><code># Calculate the number of cis pQTL proteins in each decile
cis_pQTL_proteins_per_decile &lt;- kIN_DF_TT2 %&gt;%
  filter(is_cis_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(cis_pQTL_proteins = n())


# Calculate the number of trans pQTL proteins in each decile
trans_pQTL_proteins_per_decile &lt;- kIN_DF_TT2 %&gt;%
  filter(is_trans_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(trans_pQTL_proteins = n())


# Merge the data frames
decile_summary &lt;- total_proteins_per_decile %&gt;%
  left_join(cis_pQTL_proteins_per_decile, by = &quot;Decile&quot;) %&gt;%
  left_join(trans_pQTL_proteins_per_decile, by = &quot;Decile&quot;)

# Replace NA values with 0 (for deciles with no pQTL proteins)
decile_summary$cis_pQTL_proteins[is.na(decile_summary$cis_pQTL_proteins)] &lt;- 0

decile_summary$trans_pQTL_proteins[is.na(decile_summary$trans_pQTL_proteins)] &lt;- 0

# Calculate the percentage of cis and trans pQTL proteins in each decile
decile_summary &lt;- decile_summary %&gt;%
  mutate(percentage_cis_pQTL = (cis_pQTL_proteins / total_proteins) * 100,
         percentage_trans_pQTL = (trans_pQTL_proteins / total_proteins) * 100)

# Reshape the data for plotting
decile_summary_long &lt;- decile_summary %&gt;%
  pivot_longer(cols = starts_with(&quot;percentage_&quot;), names_to = &quot;Type&quot;, values_to = &quot;Percentage&quot;)


# Replace &quot;percentage_&quot; with an empty string for better labels
decile_summary_long$Type &lt;- gsub(&quot;percentage_&quot;, &quot;&quot;, decile_summary_long$Type)

# Plot the data
ggplot(decile_summary_long, aes(x = Decile, y = Percentage, color = Type, group = Type)) +
  geom_line() +
  geom_point(size = 2) +
  labs(
    title = &quot;pQTL proportion across kIN deciles&quot;,
    x = &quot;Decile&quot;,
    y = &quot;% pQTL&quot;
  ) +
  theme_minimal() +
  scale_color_manual(values = c(&quot;cis_pQTL&quot; = &quot;darkblue&quot;, &quot;trans_pQTL&quot; = &quot;tan3&quot;)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  coord_cartesian(ylim = c(0,25))</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/pQTLs%20across%20connectivities-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>kIN_DF_TT2_DOX &lt;- kIN_DF_TT2[kIN_DF_TT2$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;), ]

kIN_DF_TT2_DOX$Decile &lt;- group_by_deciles(kIN_DF_TT2_DOX$Norm_kIN)




# Check if a protein is a cis or trans pQTL protein
kIN_DF_TT2_DOX$is_cis_pQTL &lt;- kIN_DF_TT2_DOX$Protein %in% Cis_pQTL_data_summary$Target.UniProt
kIN_DF_TT2_DOX$is_trans_pQTL &lt;- kIN_DF_TT2_DOX$Protein %in% Trans_pQTL_data_summary$Target.UniProt

# Calculate the total number of proteins in each decile
total_proteins_per_decile &lt;- kIN_DF_TT2_DOX %&gt;%
  group_by(Decile) %&gt;%
  summarise(total_proteins = n())

# Calculate the number of cis pQTL proteins in each decile
cis_pQTL_proteins_per_decile &lt;- kIN_DF_TT2_DOX %&gt;%
  filter(is_cis_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(cis_pQTL_proteins = n())

# Calculate the number of trans pQTL proteins in each decile
trans_pQTL_proteins_per_decile &lt;- kIN_DF_TT2_DOX %&gt;%
  filter(is_trans_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(trans_pQTL_proteins = n())

# Merge the data frames
decile_summary &lt;- total_proteins_per_decile %&gt;%
  left_join(cis_pQTL_proteins_per_decile, by = &quot;Decile&quot;) %&gt;%
  left_join(trans_pQTL_proteins_per_decile, by = &quot;Decile&quot;)

# Replace NA values with 0 (for deciles with no pQTL proteins)
decile_summary$cis_pQTL_proteins[is.na(decile_summary$cis_pQTL_proteins)] &lt;- 0
decile_summary$trans_pQTL_proteins[is.na(decile_summary$trans_pQTL_proteins)] &lt;- 0

# Calculate the percentage of cis and trans pQTL proteins in each decile
decile_summary &lt;- decile_summary %&gt;%
  mutate(percentage_cis_pQTL = (cis_pQTL_proteins / total_proteins) * 100,
         percentage_trans_pQTL = (trans_pQTL_proteins / total_proteins) * 100)

# Reshape the data for plotting
decile_summary_long &lt;- decile_summary %&gt;%
  pivot_longer(cols = starts_with(&quot;percentage_&quot;), names_to = &quot;Type&quot;, values_to = &quot;Percentage&quot;)


# Replace &quot;percentage_&quot; with an empty string for better labels
decile_summary_long$Type &lt;- gsub(&quot;percentage_&quot;, &quot;&quot;, decile_summary_long$Type)

# Plot the data
Plot1 &lt;- ggplot(decile_summary_long, aes(x = Decile, y = Percentage, color = Type, group = Type)) +
  geom_line() +
  geom_point(size = 2) +
  labs(
    title = &quot;DOXcorr pQTL proportion across kIN deciles&quot;,
    x = &quot;Decile&quot;,
    y = &quot;% pQTL&quot;
  ) +
  theme_minimal() +
  scale_color_manual(values = c(&quot;cis_pQTL&quot; = &quot;darkblue&quot;, &quot;trans_pQTL&quot; = &quot;tan3&quot;)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  coord_cartesian(ylim = c(2.5,25))




kIN_DF_TT2_NOT_DOX &lt;- kIN_DF_TT2[!kIN_DF_TT2$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;), ]
kIN_DF_TT2_NOT_DOX$Decile &lt;- group_by_deciles(kIN_DF_TT2_NOT_DOX$Norm_kIN)



# Check if a protein is a cis or trans pQTL protein
kIN_DF_TT2_NOT_DOX$is_cis_pQTL &lt;- kIN_DF_TT2_NOT_DOX$Protein %in% Cis_pQTL_data_summary$Target.UniProt
kIN_DF_TT2_NOT_DOX$is_trans_pQTL &lt;- kIN_DF_TT2_NOT_DOX$Protein %in% Trans_pQTL_data_summary$Target.UniProt

# Calculate the total number of proteins in each decile
total_proteins_per_decile &lt;- kIN_DF_TT2_NOT_DOX %&gt;%
  group_by(Decile) %&gt;%
  summarise(total_proteins = n())

# Calculate the number of cis pQTL proteins in each decile
cis_pQTL_proteins_per_decile &lt;- kIN_DF_TT2_NOT_DOX %&gt;%
  filter(is_cis_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(cis_pQTL_proteins = n())

# Calculate the number of trans pQTL proteins in each decile
trans_pQTL_proteins_per_decile &lt;- kIN_DF_TT2_NOT_DOX %&gt;%
  filter(is_trans_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(trans_pQTL_proteins = n())

# Merge the data frames
decile_summary &lt;- total_proteins_per_decile %&gt;%
  left_join(cis_pQTL_proteins_per_decile, by = &quot;Decile&quot;) %&gt;%
  left_join(trans_pQTL_proteins_per_decile, by = &quot;Decile&quot;)

# Replace NA values with 0 (for deciles with no pQTL proteins)
decile_summary$cis_pQTL_proteins[is.na(decile_summary$cis_pQTL_proteins)] &lt;- 0
decile_summary$trans_pQTL_proteins[is.na(decile_summary$trans_pQTL_proteins)] &lt;- 0

# Calculate the percentage of cis and trans pQTL proteins in each decile
decile_summary &lt;- decile_summary %&gt;%
  mutate(percentage_cis_pQTL = (cis_pQTL_proteins / total_proteins) * 100,
         percentage_trans_pQTL = (trans_pQTL_proteins / total_proteins) * 100)

# Reshape the data for plotting
decile_summary_long &lt;- decile_summary %&gt;%
  pivot_longer(cols = starts_with(&quot;percentage_&quot;), names_to = &quot;Type&quot;, values_to = &quot;Percentage&quot;)

# Replace &quot;percentage_&quot; with an empty string for better labels
decile_summary_long$Type &lt;- gsub(&quot;percentage_&quot;, &quot;&quot;, decile_summary_long$Type)

# Plot the data
Plot2 &lt;- ggplot(decile_summary_long, aes(x = Decile, y = Percentage, color = Type, group = Type)) +
  geom_line() +
  geom_point(size = 2) +
  labs(
    title = &quot;Non-DOXcorr pQTL proportion across kIN deciles&quot;,
    x = &quot;Decile&quot;,
    y = &quot;% pQTL&quot;
  ) +
  theme_minimal() +
  scale_color_manual(values = c(&quot;cis_pQTL&quot; = &quot;darkblue&quot;, &quot;trans_pQTL&quot; = &quot;tan3&quot;)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  coord_cartesian(ylim = c(2.5,25))

Plot1 + Plot2</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/pQTLs%20across%20connectivities-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### ALL PQTLs 
# 
kIN_DF_TT2_DOX &lt;- kIN_DF_TT2[kIN_DF_TT2$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;), ]

kIN_DF_TT2_DOX$Decile &lt;- group_by_deciles(kIN_DF_TT2_DOX$Norm_kIN)


# Check if a protein is a pQTL protein
kIN_DF_TT2_DOX$is_pQTL &lt;- kIN_DF_TT2_DOX$Protein %in% pQTL_pro_unique_exp

# Calculate the total number of proteins in each decile
total_proteins_per_decile &lt;- kIN_DF_TT2_DOX %&gt;%
  group_by(Decile) %&gt;%
  summarise(total_proteins = n())

# Calculate the number of pQTL proteins in each decile
pQTL_proteins_per_decile &lt;- kIN_DF_TT2_DOX %&gt;%
  filter(is_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(pQTL_proteins = n())

# Merge the two data frames
decile_summary &lt;- merge(total_proteins_per_decile, pQTL_proteins_per_decile, by = &quot;Decile&quot;, all.x = TRUE)

# Replace NA values with 0 (for deciles with no pQTL proteins)
decile_summary$pQTL_proteins[is.na(decile_summary$pQTL_proteins)] &lt;- 0


# Calculate the percentage of pQTL proteins in each decile
decile_summary &lt;- decile_summary %&gt;%
  mutate(percentage_pQTL = (pQTL_proteins / total_proteins) * 100)




decile_summary_DOXcorr &lt;- decile_summary
kIN_DF_TT2_NOT_DOX &lt;- kIN_DF_TT2[!kIN_DF_TT2$Modules %in% c(&quot;green&quot;, &quot;darkgreen&quot;,&quot;midnightblue&quot;,&quot;salmon&quot;,&quot;lightyellow&quot;), ]


kIN_DF_TT2_NOT_DOX$Decile &lt;- group_by_deciles(kIN_DF_TT2_NOT_DOX$Norm_kIN)


# Check if a protein is a pQTL protein
kIN_DF_TT2_NOT_DOX$is_pQTL &lt;- kIN_DF_TT2_NOT_DOX$Protein %in% pQTL_pro_unique_exp

# Calculate the total number of proteins in each decile
total_proteins_per_decile &lt;- kIN_DF_TT2_NOT_DOX %&gt;%
  group_by(Decile) %&gt;%
  summarise(total_proteins = n())

# Calculate the number of pQTL proteins in each decile
pQTL_proteins_per_decile &lt;- kIN_DF_TT2_NOT_DOX %&gt;%
  filter(is_pQTL) %&gt;%
  group_by(Decile) %&gt;%
  summarise(pQTL_proteins = n())

# Merge the two data frames
decile_summary &lt;- merge(total_proteins_per_decile, pQTL_proteins_per_decile, by = &quot;Decile&quot;, all.x = TRUE)

# Replace NA values with 0 (for deciles with no pQTL proteins)
decile_summary$pQTL_proteins[is.na(decile_summary$pQTL_proteins)] &lt;- 0

# Calculate the percentage of pQTL proteins in each decile
decile_summary &lt;- decile_summary %&gt;%
  mutate(percentage_pQTL = (pQTL_proteins / total_proteins) * 100)
 decile_summary_NOTDOX &lt;- decile_summary
 decile_summary_DOXcorr$Type &lt;- c(&quot;DOX-corr.&quot;)
 decile_summary_NOTDOX$Type &lt;- c(&quot;Not-DOX-corr.&quot;)
 decile_summary_ALL &lt;- rbind(decile_summary_DOXcorr,decile_summary_NOTDOX )
  
 
 
 
# Define the colors for each type
type_colors &lt;- c(&quot;DOX-corr.&quot; = &quot;red&quot;, &quot;Not-DOX-corr.&quot; = &quot;blue&quot;)
# Plot the data
ggplot(decile_summary_ALL, aes(x = Decile, y = percentage_pQTL, color = Type, group = Type)) +
  geom_line() +
  geom_point(size = 2) +
  labs(
    title = &quot;Percentage of pQTL Proteins Across Deciles&quot;,
    x = &quot;Decile&quot;,
    y = &quot;Percentage of pQTL Proteins&quot;
  ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_color_manual(values = type_colors)</code></pre>
<p><img src="figure/Johnson_DOX_24_5.Rmd/pQTLs%20across%20connectivities-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>decile_summary_ALL</code></pre>
<pre><code>   Decile total_proteins pQTL_proteins percentage_pQTL          Type
1      D1            202            33       16.336634     DOX-corr.
2     D10            202            16        7.920792     DOX-corr.
3      D2            202            35       17.326733     DOX-corr.
4      D3            201            42       20.895522     DOX-corr.
5      D4            202            30       14.851485     DOX-corr.
6      D5            202            26       12.871287     DOX-corr.
7      D6            201            36       17.910448     DOX-corr.
8      D7            202            46       22.772277     DOX-corr.
9      D8            201            21       10.447761     DOX-corr.
10     D9            202            20        9.900990     DOX-corr.
11     D1            217            19        8.755760 Not-DOX-corr.
12    D10            216            46       21.296296 Not-DOX-corr.
13     D2            216            26       12.037037 Not-DOX-corr.
14     D3            216            36       16.666667 Not-DOX-corr.
15     D4            216            35       16.203704 Not-DOX-corr.
16     D5            216            49       22.685185 Not-DOX-corr.
17     D6            216            41       18.981481 Not-DOX-corr.
18     D7            216            55       25.462963 Not-DOX-corr.
19     D8            216            38       17.592593 Not-DOX-corr.
20     D9            216            52       24.074074 Not-DOX-corr.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scales_1.2.1                           
 [2] ReactomePA_1.40.0                      
 [3] impute_1.70.0                          
 [4] WGCNA_1.72-1                           
 [5] fastcluster_1.2.3                      
 [6] dynamicTreeCut_1.63-1                  
 [7] BioNERO_1.4.2                          
 [8] reshape2_1.4.4                         
 [9] ggridges_0.5.4                         
[10] biomaRt_2.52.0                         
[11] ggvenn_0.1.10                          
[12] UpSetR_1.4.0                           
[13] DOSE_3.22.1                            
[14] variancePartition_1.26.0               
[15] clusterProfiler_4.4.4                  
[16] pheatmap_1.0.12                        
[17] qvalue_2.28.0                          
[18] Homo.sapiens_1.3.1                     
[19] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[20] org.Hs.eg.db_3.15.0                    
[21] GO.db_3.15.0                           
[22] OrganismDbi_1.38.1                     
[23] GenomicFeatures_1.48.4                 
[24] AnnotationDbi_1.58.0                   
[25] cluster_2.1.4                          
[26] ggfortify_0.4.16                       
[27] lubridate_1.9.2                        
[28] forcats_1.0.0                          
[29] stringr_1.5.0                          
[30] dplyr_1.1.2                            
[31] purrr_1.0.2                            
[32] readr_2.1.4                            
[33] tidyr_1.3.0                            
[34] tibble_3.2.1                           
[35] ggplot2_3.4.3                          
[36] tidyverse_2.0.0                        
[37] RColorBrewer_1.1-3                     
[38] RUVSeq_1.30.0                          
[39] edgeR_3.38.4                           
[40] limma_3.52.4                           
[41] EDASeq_2.30.0                          
[42] ShortRead_1.54.0                       
[43] GenomicAlignments_1.32.1               
[44] SummarizedExperiment_1.26.1            
[45] MatrixGenerics_1.8.1                   
[46] matrixStats_1.0.0                      
[47] Rsamtools_2.12.0                       
[48] GenomicRanges_1.48.0                   
[49] Biostrings_2.64.1                      
[50] GenomeInfoDb_1.32.4                    
[51] XVector_0.36.0                         
[52] IRanges_2.30.1                         
[53] S4Vectors_0.34.0                       
[54] BiocParallel_1.30.4                    
[55] Biobase_2.56.0                         
[56] BiocGenerics_0.42.0                    
[57] workflowr_1.7.1                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         rtracklayer_1.56.1     minet_3.54.0          
  [4] R.methodsS3_1.8.2      coda_0.19-4            bit64_4.0.5           
  [7] knitr_1.43             aroma.light_3.26.0     DelayedArray_0.22.0   
 [10] R.utils_2.12.2         rpart_4.1.19           data.table_1.14.8     
 [13] hwriter_1.3.2.1        KEGGREST_1.36.3        RCurl_1.98-1.12       
 [16] doParallel_1.0.17      generics_0.1.3         preprocessCore_1.58.0 
 [19] callr_3.7.3            RhpcBLASctl_0.23-42    RSQLite_2.3.1         
 [22] shadowtext_0.1.2       bit_4.0.5              tzdb_0.4.0            
 [25] enrichplot_1.16.2      xml2_1.3.5             httpuv_1.6.11         
 [28] viridis_0.6.4          xfun_0.40              hms_1.1.3             
 [31] jquerylib_0.1.4        evaluate_0.21          promises_1.2.1        
 [34] fansi_1.0.4            restfulr_0.0.15        progress_1.2.2        
 [37] caTools_1.18.2         dbplyr_2.3.3           htmlwidgets_1.6.2     
 [40] igraph_1.5.1           DBI_1.1.3              ggnewscale_0.4.9      
 [43] backports_1.4.1        annotate_1.74.0        aod_1.3.2             
 [46] deldir_1.0-9           vctrs_0.6.3            abind_1.4-5           
 [49] cachem_1.0.8           withr_2.5.0            ggforce_0.4.1         
 [52] checkmate_2.2.0        treeio_1.20.2          prettyunits_1.1.1     
 [55] ape_5.7-1              lazyeval_0.2.2         crayon_1.5.2          
 [58] genefilter_1.78.0      labeling_0.4.2         pkgconfig_2.0.3       
 [61] tweenr_2.0.2           nlme_3.1-163           nnet_7.3-19           
 [64] rlang_1.1.1            lifecycle_1.0.3        downloader_0.4        
 [67] filelock_1.0.2         BiocFileCache_2.4.0    rprojroot_2.0.3       
 [70] polyclip_1.10-4        graph_1.74.0           Matrix_1.5-4.1        
 [73] aplot_0.2.0            NetRep_1.2.7           boot_1.3-28.1         
 [76] base64enc_0.1-3        GlobalOptions_0.1.2    whisker_0.4.1         
 [79] processx_3.8.2         png_0.1-8              viridisLite_0.4.2     
 [82] rjson_0.2.21           bitops_1.0-7           getPass_0.2-2         
 [85] R.oo_1.25.0            ggnetwork_0.5.12       KernSmooth_2.23-22    
 [88] blob_1.2.4             shape_1.4.6            jpeg_0.1-10           
 [91] gridGraphics_0.5-1     reactome.db_1.81.0     graphite_1.42.0       
 [94] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.8            
 [97] gplots_3.1.3           zlibbioc_1.42.0        compiler_4.2.0        
[100] scatterpie_0.2.1       BiocIO_1.6.0           clue_0.3-64           
[103] intergraph_2.0-3       lme4_1.1-34            cli_3.6.1             
[106] patchwork_1.1.3        ps_1.7.5               htmlTable_2.4.1       
[109] Formula_1.2-5          mgcv_1.9-0             MASS_7.3-60           
[112] tidyselect_1.2.0       stringi_1.7.12         highr_0.10            
[115] yaml_2.3.7             GOSemSim_2.22.0        locfit_1.5-9.8        
[118] latticeExtra_0.6-30    ggrepel_0.9.3          sass_0.4.7            
[121] fastmatch_1.1-4        tools_4.2.0            timechange_0.2.0      
[124] parallel_4.2.0         circlize_0.4.15        rstudioapi_0.15.0     
[127] foreign_0.8-84         foreach_1.5.2          git2r_0.32.0          
[130] gridExtra_2.3          farver_2.1.1           ggraph_2.1.0          
[133] digest_0.6.33          BiocManager_1.30.22    networkD3_0.4         
[136] Rcpp_1.0.11            broom_1.0.5            later_1.3.1           
[139] httr_1.4.7             ComplexHeatmap_2.12.1  GENIE3_1.18.0         
[142] Rdpack_2.5             colorspace_2.1-0       XML_3.99-0.14         
[145] fs_1.6.3               splines_4.2.0          statmod_1.5.0         
[148] yulab.utils_0.0.8      RBGL_1.72.0            tidytree_0.4.5        
[151] graphlayouts_1.0.0     ggplotify_0.1.2        xtable_1.8-4          
[154] jsonlite_1.8.7         nloptr_2.0.3           ggtree_3.4.4          
[157] tidygraph_1.2.3        ggfun_0.1.2            R6_2.5.1              
[160] Hmisc_5.1-0            pillar_1.9.0           htmltools_0.5.6       
[163] glue_1.6.2             fastmap_1.1.1          minqa_1.2.5           
[166] codetools_0.2-19       fgsea_1.22.0           utf8_1.2.3            
[169] sva_3.44.0             lattice_0.21-8         bslib_0.5.1           
[172] network_1.18.1         pbkrtest_0.5.2         curl_5.0.2            
[175] gtools_3.9.4           interp_1.1-4           survival_3.5-7        
[178] statnet.common_4.9.0   rmarkdown_2.24         munsell_0.5.0         
[181] GetoptLong_1.0.5       DO.db_2.9              GenomeInfoDbData_1.2.8
[184] iterators_1.0.14       gtable_0.3.4           rbibutils_2.2.15      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
