---
title: "Johnson_DOX_24_8"
author: "Omar Johnson"
date: "2024-07-22"
output: html_document
---

# Load Libraries 
```{r Libraries, include=FALSE}
library(EDASeq)
library(RUVSeq)
library(RColorBrewer)
library(edgeR)
library(limma)
library(Biobase)
library(SummarizedExperiment)
library(tidyverse) 
library(ggfortify)
library(cluster)
library(edgeR)
library(limma)
library(Homo.sapiens)
library(BiocParallel)
library(qvalue)
library(pheatmap)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(variancePartition)
library(DOSE)
library(UpSetR)
library(ggvenn)
library(biomaRt)
library(ggridges)
library(reshape2)
library(BioNERO)
library(WGCNA)
library(impute)
library(dynamicTreeCut)
library(ReactomePA)
library(scales)
library(ggraph)
library(igraph)
library(tidygraph)
```

# Read in Data 
```{r Read in data, include=FALSE}

# 1. RUVg Corrected data across all 10 samples that has been log2 transformed. It has not yet been quantile normalized. 
 RUVg_Log2_quantnormalized_all10samples <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/RUVIII_10samples_log2_notquantilenormalized.csv", header = TRUE, row.names = 1)
RUVg_Log2_quantnormalized_all10samples <- RUVg_Log2_quantnormalized_all10samples^2
 

# 2. Toptable correspondding to the Diff.Abundance test from #1 
toptable_summary <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/Toptable_summary_RUVIII.csv", header = TRUE, row.names = 1)


# 3. Meta data for all 10 samples in our study. 
Meta <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/Meta.csv", header = TRUE, row.names = 1)


Toptable_Modules <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/SUPP_Table_2.csv", header = TRUE)

colnames(Toptable_Modules) <- c("X", "Protein","kTotal", "kWithin",        "kOut", "kDiff", "logFC", "AveExpr", "t", "P.Value","adj.P.Val" ,"B" , "threshold_P", "Modules", "DE_or_Not", "Norm_kIN", "Norm_kOut", "logFC.y",     "AveExpr.y", "t.y", "P.Value.y", "adj.P.Val.y", "B.y" , "threshold_P.y",    "Modules.y", "DE_or_Not.y", "Is_DA", "Is_DOXcorrelated", "Is_Hub", "Is_Cis_pQTL", "Is_Trans_pQTL", "Is_pQTL", "pLI_assigned", "pLI_Mut.Intolerant", "pLI_Mut.Tolerant", "Is_Druggable", "Is_CVD_protein",    "Is_CVD_PPI_protein")

New_RNA_PRO_DF_2 <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/New_RNA_PRO_DF.csv", header = TRUE, row.names = 1)


New_RNA_PRO_DF_3 <- merge(Toptable_Modules, New_RNA_PRO_DF_2, by.x = "Protein", by.y = "uniprotswissprot")


hubs <- read.csv( file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/hubs.csv", header = TRUE)


GO_results_DOX <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/GO_results.csv", header = TRUE)


HPA_General3 <- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/HPA_General3_test.RData")

HPA_General4_test <- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/HPA_General4.RData")

HPA_General2_test <- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/HPA_General2.RData")


dbd_uniprot_list_test<- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/dbd_uniprot_list.RData")


RBP_pros_uniprot_3 <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/RBP_pros_uniprot_3.csv", header = TRUE)

# Set the path to the folder containing the .tsv files
folder_path <- "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/HPA_Metabolism" 

EnzymeTable <- read.csv(file ="/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/HPA_Metabolism_Labels/HPA_Metabo_labels.csv" , header = TRUE)

TF_UNIPROT_ENSEMBL_2_2 <- read.csv(file ="/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/TF_UNIPROT_ENSEMBL_2.csv" , header = TRUE)

File_path_1 <- "/Users/omarjohnson/Downloads/EFO_0003777_associations_export.tsv"

File_path_2 <-"/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/pLI/Genes_pLI_Scores.csv"

PPI_Name_Key <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/PPI_Name_Key.csv", header = TRUE)

Enrichment_DF_ALL_GWASuni <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/Enrichment_DF_ALL_GWASuni.csv", header = TRUE)

CVD_net <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/PPI_Functionally_annotated/PPI_WGCNA_DOX_GWASproCVD_annotated.csv" , header = TRUE)

TT_Annotated <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/PPI_Functionally_annotated/PPI_Functionally_annotated.csv" , header = TRUE)

Name_Key <- read.csv(file = "/Users/omarjohnson/Downloads/idmapping_2024_07_01.csv", header = TRUE)


```

# Functions 
```{r Functions}
perform_module_comparisons_mutexc_2 <- function(df, module_col, value_col) {
  # Ensure the necessary columns exist
  if (!(module_col %in% names(df) && value_col %in% names(df))) {
    stop("Specified columns do not exist in the dataframe.")
  }

  # Get a list of all unique modules
  modules <- unique(df[[module_col]])

  # Initialize an empty list to store combined data frames
  combined_df_list <- list()

  # Initialize an empty dataframe to store results
  results <- data.frame(Module1 = character(),
                        Module2 = character(),
                        WilcoxPValue = numeric(),
                        stringsAsFactors = FALSE)

  # Loop through each module
  for (module in modules) {
    # Data for the current module
    current_data <- df %>% filter(!!sym(module_col) == module) %>% 
                    mutate(Group = as.character(module))

    # Data for all other modules
    other_data <- df %>% filter(!!sym(module_col) != module) %>% 
                    mutate(Group = paste("Not", module, sep=""))

    # Combine current module data with other module data
    combined_data <- rbind(current_data, other_data)

    # Add the combined data to the list
    combined_df_list[[module]] <- combined_data

    # Perform the Wilcoxon test
    test_result <- wilcox.test(current_data[[value_col]], other_data[[value_col]])

    # Add the results to the dataframe
    results <- rbind(results, data.frame(Module1 = module,
                                         Module2 = "Others",
                                         WilcoxPValue = test_result$p.value))
  }

  return(list("results" = results, "combined_data" = combined_df_list))
}


perform_module_disease_analysis_genes_3 <- function(toptable, diseaseGenes) {
  # Prepare an empty list to collect results
  results <- list()
  
  # Ensure 'Modules' and 'hgnc_symbol' columns exist in 'toptable'
  if(!"Modules" %in% names(toptable)) {
    stop("Column 'Modules' not found in the 'toptable'.")
  }
  if(!"hgnc_symbol" %in% names(toptable)) {
    stop("Column 'hgnc_symbol' not found in the 'toptable'.")
  }
  
  # Filter disease genes to include only those that are expressed in toptable
  expressedDiseaseGenes <- lapply(diseaseGenes, function(genes) {
    intersect(genes, toptable$hgnc_symbol)
  })
  
  # Loop through each module
  modules <- unique(toptable$Modules)
  for (module in modules) {
    # Get the genes in the module
    moduleGenes <- toptable$hgnc_symbol[toptable$Modules == module]
    
    # Loop through each disease gene set
    for (diseaseName in names(expressedDiseaseGenes)) {
      # Find the intersecting genes between the module and the expressed disease genes
      diseaseModuleIntersect <- intersect(moduleGenes, expressedDiseaseGenes[[diseaseName]])
      
      # Calculate elements for the contingency table
      numIntersect = length(diseaseModuleIntersect)
      numInModuleNotDisease = length(moduleGenes) - numIntersect
      numInDiseaseNotModule = length(expressedDiseaseGenes[[diseaseName]]) - numIntersect
      numInNeither = nrow(toptable) - (numIntersect + numInModuleNotDisease + numInDiseaseNotModule)
      
      # Build the contingency table
      table <- matrix(c(
        numIntersect, # Both in disease list and module
        numInModuleNotDisease, # In module but not disease list
        numInDiseaseNotModule, # In disease list but not module
        numInNeither # In neither list
      ), nrow = 2, byrow = TRUE)
      
      # Perform chi-squared test and Fisher's exact test with error handling
      chiSqTestResult <- tryCatch({
        chisq.test(table, correct = TRUE)
      }, error = function(e) {
        list(p.value = NA)
      }, warning = function(w) {
        list(p.value = NA)
      })
      
      fisherTestResult <- tryCatch({
        fisher.test(table)
      }, error = function(e) {
        list(p.value = NA)
      }, warning = function(w) {
        list(p.value = NA)
      })
      
      # Calculate percent overlap, handle division by zero
      percentOverlap <- if (length(moduleGenes) > 0) {
        (numIntersect / length(expressedDiseaseGenes[[diseaseName]])) * 100
      } else {
        0
      }
      
      # Convert intersecting genes to a single character string
      intersectingGenesStr <- if (numIntersect > 0) {
        paste(diseaseModuleIntersect, collapse = ";")
      } else {
        ""  # Use an empty string to indicate no intersection
      }
      
      # Append to results list
      results[[paste(module, diseaseName, sep = "_")]] <- data.frame(
        Modules = module,
        Disease = diseaseName,
        ChiSqPValue = chiSqTestResult$p.value,
        FisherPValue = fisherTestResult$p.value,
        PercentOverlap = percentOverlap,
        OddsRatio = fisherTestResult$estimate, 
        IntersectingGenes = intersectingGenesStr
      )
    }
  }
  
  # Combine results into a single data frame
  results_df <- do.call(rbind, results)
  return(results_df)
}



# Function assignment 
perform_fisher_test_FP <- function(vec1, vec2, vec1_name, vec2_name, plot = FALSE) {
  # Create labeled factors for vec1 and vec2
  vec1_label <- factor(vec1, labels = c(paste0("Not", vec1_name), paste0("Is", vec1_name)))
  vec2_label <- factor(vec2, labels = c(paste0("Not", vec2_name), paste0("Is", vec2_name)))

  # Create contingency table with labeled factors
  table <- table(vec1_label, vec2_label)

  # Perform Fisher's exact test
  test_result <- fisher.test(table)
  p_value <- test_result$p.value
OR <- test_result$estimate
CI <- test_result$conf.int

  # Prepare result
  result <- list(
    ContingencyTable = table,
    PValue = p_value, 
    Odds_ratio = test_result$estimate,
    Confidence_Interval = test_result$conf.int
  )

  # Generate plot if required
  if (plot) {
    # Convert table to data frame for ggplot
    table_df <- as.data.frame(as.table(table))
    colnames(table_df) <- c("vec1_label", "vec2_label", "Freq")

    # Calculate totals for each vec1_label
    totals <- aggregate(Freq ~ vec1_label, data = table_df, sum)

    # Merge totals with table_df and calculate percentages
    table_df <- merge(table_df, totals, by = "vec1_label", all.x = TRUE)
    table_df$Percentage <- with(table_df, Freq.x / Freq.y * 100)
    table_df$Group <- table_df$vec2_label

    # Stacked bar chart
    p <- ggplot(table_df, aes(x = vec1_label, y = Percentage, fill = Group)) +
      geom_bar(stat = "identity", position = "stack") +  # Adjust position to "stack"
      facet_wrap(~ vec1_label) +
      theme_minimal() +
      labs(x = vec1_name, y = "Percentage", fill = vec2_name, title = paste("")) +
      theme(axis.text.x = element_text(angle = 45, hjust = 1))

    result$Plot <- p
  }

  return(result)
}


group_by_deciles <- function(x) {
  deciles <- cut(x, 
                 breaks = quantile(x, probs = seq(0, 1, by = 0.1), na.rm = TRUE), 
                 include.lowest = TRUE, 
                 labels = paste0("D", 1:10))
  return(deciles)
}
```

# 8B-Odds of being CVD pro
```{r }
BigGWASsumstat <- read_tsv(file = File_path_1)


BigGWASsumstat_sep <- separate_rows(BigGWASsumstat, mappedGenes, sep = ",")
BigGWASsumstat_sep %>% nrow()
GWAS_Traits_to_remove <- BigGWASsumstat_sep[BigGWASsumstat_sep$mappedGenes == "-", ]$traitName %>% unique()


BigGWASsumstat_sep_filt <- BigGWASsumstat_sep[!BigGWASsumstat_sep$traitName %in% GWAS_Traits_to_remove, ]
CVD_GWAS_genes <- BigGWASsumstat_sep_filt$mappedGenes %>% unique()

GWAS_pros <- PPI_Name_Key[PPI_Name_Key$hgnc_symbol %in% CVD_GWAS_genes, ]$uniprotswissprot



pQTL_data_summary_merged <- Toptable_Modules



pQTL_data_summary_merged <- pQTL_data_summary_merged %>%
  mutate(Is_hub = if_else(Toptable_Modules$Protein %in% hubs$Gene, 1, 0))


pQTL_data_summary_merged <- pQTL_data_summary_merged %>%
  mutate(Is_DA = if_else(P.Value < 0.05, 1, 0))

pQTL_data_summary_merged <- pQTL_data_summary_merged %>%
  mutate(Is_GWAS = if_else(Protein %in% GWAS_pros, 1, 0))


pQTL_data_summary_merged <- pQTL_data_summary_merged %>%
  mutate(Is_DOXcorr = if_else((Modules %in% c("green","darkgreen","midnightblue","salmon","lightyellow")), 1, 0))


result_pLI_DOXcorr <- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged$Is_DOXcorr, vec2 = pQTL_data_summary_merged$Is_GWAS, vec1_name = "DAP", vec2_name = "GWAS_Pro",  plot = FALSE)


result_pLI_HUB <- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged$Is_hub, vec2 = pQTL_data_summary_merged$Is_GWAS, vec1_name = "Hub", vec2_name = "GWAS_Pro",  plot = FALSE)


pQTL_data_summary_merged_hubs <- pQTL_data_summary_merged[pQTL_data_summary_merged$Is_hub == TRUE, ]


result_pLI_DOX_corr_hub <- perform_fisher_test_FP(vec1 = pQTL_data_summary_merged_hubs$Is_DOXcorr, vec2 = pQTL_data_summary_merged_hubs$Is_GWAS, vec1_name = "DOXcorr hub", vec2_name = "GWAS_Pro",  plot = TRUE)


FP_List <- list(result_pLI_DOX_corr_hub, result_pLI_HUB, result_pLI_DOXcorr)

FP_DF <- data.frame(
  Odds_ratio = numeric(length(FP_List)),
  Lower_CI = numeric(length(FP_List)),
  Upper_CI = numeric(length(FP_List)), 
  Pval = numeric(length(FP_List))
)

for (i in 1:length(FP_List)) {
  FP_DF$Odds_ratio[i] <- FP_List[[i]]$Odds_ratio
  FP_DF$Lower_CI[i] <- FP_List[[i]]$Confidence_Interval[1]
  FP_DF$Upper_CI[i] <- FP_List[[i]]$Confidence_Interval[2]
  FP_DF$Pval[i] <- FP_List[[i]]$PValue
}

# Add row names for the labels in the forest plot
FP_DF$Label <- c("DOXcorrhub", "hub", "DOXcorr")


FP_DF$Label <- factor(FP_DF$Label, levels = rev(c( "hub", "DOXcorr", "DOXcorrhub")))

ggplot(FP_DF, aes(x = Label, y = Odds_ratio, ymin = Lower_CI, ymax = Upper_CI)) +
  geom_pointrange() +
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +
  coord_flip() +
  labs(
    title = "CVD risk protein enrichment across network proteins",
    x = "",
    y = "Odds of being CVD protein (95% CI)"
  ) +
  theme_classic()
  
```

# 8D-Odds of being CVD-PPI
```{r }

Enrichment_DF_ALL_GWASuni <- Enrichment_DF_ALL_GWASuni %>%
  mutate(Is_DOXcorr = if_else(Enrichment_DF_ALL_GWASuni$Modules %in% c("green","darkgreen","midnightblue","salmon","lightyellow"), 1, 0))

FP_DOXcorr_PPI <- perform_fisher_test_FP(vec1 = Enrichment_DF_ALL_GWASuni$Is_GWAS_PPI, vec2 = Enrichment_DF_ALL_GWASuni$Is_DOXcorr, vec1_name = "PPI with GWAS protein", vec2_name = "DOXcorr.Hub",  plot = TRUE)


FP_Hub_PPI <- perform_fisher_test_FP(vec1 = Enrichment_DF_ALL_GWASuni$Is_GWAS_PPI, vec2 = Enrichment_DF_ALL_GWASuni$Is_Hub, vec1_name = "PPI with GWAS protein", vec2_name = "DAPs",  plot = TRUE)



Enrichment_DF_ALL_GWASuni_hub <- Enrichment_DF_ALL_GWASuni[Enrichment_DF_ALL_GWASuni$Is_Hub == 1, ]

FP_DOXcorr_Hub_PPI <- perform_fisher_test_FP(vec1 = Enrichment_DF_ALL_GWASuni_hub$Is_GWAS_PPI, vec2 = Enrichment_DF_ALL_GWASuni_hub$Is_DOXcorr.Hub, vec1_name = "PPI with GWAS protein", vec2_name = "DOX Corr. hub",  plot = TRUE)


FP_List <- list(FP_DOXcorr_Hub_PPI, FP_Hub_PPI, FP_DOXcorr_PPI)

FP_DF <- data.frame(
  Odds_ratio = numeric(length(FP_List)),
  Lower_CI = numeric(length(FP_List)),
  Upper_CI = numeric(length(FP_List)), 
  Pval = numeric(length(FP_List))
)

for (i in 1:length(FP_List)) {
  FP_DF$Odds_ratio[i] <- FP_List[[i]]$Odds_ratio
  FP_DF$Lower_CI[i] <- FP_List[[i]]$Confidence_Interval[1]
  FP_DF$Upper_CI[i] <- FP_List[[i]]$Confidence_Interval[2]
  FP_DF$Pval[i] <- FP_List[[i]]$PValue
}

# Add row names for the labels in the forest plot
FP_DF$Label <- c("DOXcorrhub", "hub", "DOXcorr")


FP_DF$Label <- factor(FP_DF$Label, levels = rev(c("hub", "DOXcorr", "DOXcorrhub")))

ggplot(FP_DF, aes(x = Label, y = Odds_ratio, ymin = Lower_CI, ymax = Upper_CI)) +
  geom_pointrange() +
  geom_hline(yintercept = 1, linetype = "dashed", color = "red") +
  coord_flip() +
  labs(
    title = "CVD risk protein physical interactor enrichment",
    x = "",
    y = "Odds of being physical protein interactor with CVD proteins (95% CI)"
  ) +
  theme_minimal()+
  ylim(c(0,16))
  
```

# 8C-CVD-pro vs CVD-PPI pLI scores
```{r }

pLI_Data <- read.csv(file = File_path_2, header = TRUE)
pLI_Data_sub <- merge(pLI_Data, New_RNA_PRO_DF_3, by.x = "gene", by.y = "hgnc_symbol")
pLI_Data_sub2 <- pLI_Data_sub[,c(1,2,3)]

Toptable_Modules_pLI <- merge(Toptable_Modules, pLI_Data_sub2, by.x = "Protein" , by.y = "Protein")

# CVD protein pLI distribution 
Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_protein == 1, ]$pLI %>% hist()

# CVD protein-PPI pLI distrbution
Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_PPI_protein == 1, ]$pLI %>% hist()



wilcox.test(Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_protein == 1, ]$pLI, Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_PPI_protein == 1, ]$pLI)

pLI_Boxplot <- data.frame(
  values = c(Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_protein == 1, ]$pLI,
             
             Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_PPI_protein == 1, ]$pLI),
  
  
  group = factor(c(rep("GWAS Pro", length(Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_protein == 1, ]$pLI)), rep("GWAS Pro-Interactors", length(Toptable_Modules_pLI[Toptable_Modules_pLI$Is_CVD_PPI_protein == 1, ]$pLI))))
)



# Create boxplot
ggplot(pLI_Boxplot, aes(x = group, y = values)) +
  geom_boxplot() +
  labs(x = "Group", y = "Mutation intolerance (pLI)") +
  theme_minimal()
```

# 8E- CVD pro PPI network
```{r }
gwas_proteins <- Toptable_Modules[Toptable_Modules$Is_CVD_protein == 1, ]$Protein

##### Net-1####
# Create the igraph object from the specific columns
g <- graph_from_data_frame(d = CVD_net[, c("query_term1", "query_term2", "Weight")], directed = FALSE)

# Annotate the graph with additional information
V(g)$is_hub <- V(g)$name %in% hubs$Gene
V(g)$is_dox_correlated <- V(g)$name %in% Toptable_Modules[Toptable_Modules$Modules %in% c("green", "darkgreen", "midnightblue", "salmon", "lightyellow"), ]$Protein
V(g)$is_cvd_pro <- V(g)$name %in% gwas_proteins

# Set vertex size based on is_hub
V(g)$size <- ifelse(V(g)$is_hub, 10, 5)  # Hubs will be larger

# Set vertex shape based on is_cvd_pro
V(g)$shape <- ifelse(V(g)$is_cvd_pro, "square", "circle")  # CVD proteins will be square

# Set vertex color based on is_dox_correlated
V(g)$color <- ifelse(V(g)$is_dox_correlated, "red", "blue")  # Dox correlated proteins will be red, others blue

# Option1 
# Plot the graph
plot(g, vertex.label = NA) 





##### Net-2 #####

# Create the igraph object from the specific columns
g <- graph_from_data_frame(d = CVD_net[, c("query_term1", "query_term2", "Weight")], directed = FALSE)

# Convert igraph object to tidygraph object
tg <- as_tbl_graph(g)

# Annotate the graph with additional information
tg <- tg %>%
  mutate(is_hub = name %in% hubs$Gene,
         is_dox_correlated = name %in% Toptable_Modules[Toptable_Modules$Modules %in% c("green", "darkgreen", "midnightblue", "salmon", "lightyellow"), ]$Protein,
         is_cvd_pro = name %in% gwas_proteins,
         size = ifelse(is_hub, 10, 5),  # Hubs will be larger
         shape = ifelse(is_cvd_pro, "square", "circle"),  # CVD proteins will be square
         color = ifelse(is_dox_correlated, "red", "blue"))  # Dox correlated proteins will be red, others blue

# Plot
ggraph(tg, layout = "kk") +
  geom_edge_link0(aes(edge_color = Weight, edge_width = Weight), show.legend = TRUE) +
  geom_node_point(aes(size = size, shape = shape, color = color), show.legend = TRUE) +
  geom_node_text(aes(label = name), fontface = "bold") +
  scale_edge_color_continuous(low = "white", high = "black") +
  scale_edge_width(range = c(0.1, .2)) +
  scale_size_continuous(range = c(5, 10)) +  # Ensure the size range is the same as in the annotations
  scale_shape_manual(values = c("circle" = 16, "square" = 15)) +  # Use specific shapes
  scale_color_manual(values = c("red", "blue")) +  # Ensure the colors are used as in the annotations
  theme_graph() +
  coord_fixed()


#### Net-3 ##### 

# Convert igraph object to tidygraph object
tg <- as_tbl_graph(g)

# Annotate the graph with additional information
tg <- tg %>%
  mutate(is_hub = name %in% hubs$Gene,
         is_dox_correlated = name %in% Toptable_Modules[Toptable_Modules$Modules %in% c("green", "darkgreen", "midnightblue", "salmon", "lightyellow"), ]$Protein,
         is_cvd_pro = name %in% gwas_proteins,
         size = ifelse(is_hub, 10, 5),  # Hubs will be larger
         shape = ifelse(is_cvd_pro, "square", "circle"),  # CVD proteins will be square
         color = ifelse(is_dox_correlated, "red", "blue"))  # Dox correlated proteins will be red, others blue

# Define edge color based on the sign of the weight
tg <- tg %>%
  activate(edges) %>%
  mutate(edge_color = ifelse(Weight > 0, "darkgreen", "darkred"))

# Plot the graph using ggraph
ggraph(tg, layout = "kk") +
  geom_edge_link(aes(edge_color = edge_color, edge_width = abs(Weight)), show.legend = TRUE) +
  geom_node_point(aes(size = size, shape = shape, color = color), show.legend = TRUE) +
  geom_node_text(aes(label = name), fontface = "bold") +
  scale_edge_color_manual(values = c("darkgreen" = "darkgreen", "darkred" = "darkred")) +
  scale_edge_width(range = c(0.5, 2)) +  # Thicker edges
  scale_size_continuous(range = c(5, 10)) +  # Ensure the size range is the same as in the annotations
  scale_shape_manual(values = c("circle" = 16, "square" = 15)) +  # Use specific shapes
  scale_color_manual(values = c("red" = "red", "blue" = "blue")) +  # Ensure the colors are used as in the annotations
  theme_graph() +
  coord_fixed()




##### Net-4 #### 
# Network with protein names 

Name_Key <- read.csv(file = "/Users/omarjohnson/Downloads/idmapping_2024_07_01.csv", header = TRUE)

Toptable_Modules_key <- merge(New_RNA_PRO_DF_3,Name_Key, by.x ="Protein" , by.y = "From" )

# Create the igraph object from the specific columns
g <- graph_from_data_frame(d = CVD_net[, c("query_term1", "query_term2", "Weight")], directed = FALSE)


# Create lookup vectors for Protein to Gene name conversion
protein_to_gene <- setNames(New_RNA_PRO_DF_3$hgnc_symbol, New_RNA_PRO_DF_3$Protein)


# Replace protein names in query_term1 and query_term2 with gene names using recode
CVD_net_2 <- CVD_net %>%
  mutate(Gene1 = recode(query_term1, !!!protein_to_gene),
         Gene2 = recode(query_term2, !!!protein_to_gene))


# Select the columns with Gene names and Weight
CVD_net_3 <- CVD_net_2 %>%
  select(Gene1, Gene2, Weight)




# Create the igraph object from the specific columns
g <- graph_from_data_frame(d = CVD_net_3, directed = FALSE)

# Convert igraph object to tidygraph object
tg <- as_tbl_graph(g)




New_RNA_PRO_DF_3_hubs <- merge(New_RNA_PRO_DF_3, hubs, by.x = "Protein", by.y = "Gene")




# Annotate the graph with additional information
tg <- tg %>%
  mutate(is_hub = name %in% New_RNA_PRO_DF_3_hubs$hgnc_symbol,
         is_dox_correlated = name %in% New_RNA_PRO_DF_3[New_RNA_PRO_DF_3$Modules %in% c("green", "darkgreen", "midnightblue", "salmon", "lightyellow"), ]$hgnc_symbol,
         is_cvd_pro = name %in% CVD_GWAS_genes,
         size = ifelse(is_hub, 10, 5),  # Hubs will be larger
         shape = ifelse(is_cvd_pro, "square", "circle"), 
          # CVD proteins will be square
         color = ifelse(is_dox_correlated, "red", "blue"))  # CVD proteins will be square




ggraph(tg, layout = "kk") +
  geom_edge_link0(aes(edge_color = Weight, edge_width = Weight), show.legend = TRUE) +
  geom_node_point(aes(size = size, shape = shape, color = color), show.legend = TRUE) +
  geom_node_text(aes(label = name), fontface = "bold") +
  scale_edge_color_continuous(low = "white", high = "black") +
  scale_edge_width(range = c(0.1, .2)) +
  scale_size_continuous(range = c(5, 10)) +  # Ensure the size range is the same as in the annotations
  scale_shape_manual(values = c("circle" = 16, "square" = 15)) +  # Use specific shapes
  scale_color_manual(values = c("red", "blue")) +  # Ensure the colors are used as in the annotations
  theme_graph() +
  coord_fixed()


```