---
title: "Johnson_DOX_24_4"
author: "Omar Johnson"
date: "2024-07-18"
output: html_document
---

# Load Libraries 
```{r Libraries, include=FALSE}
library(EDASeq)
library(RUVSeq)
library(RColorBrewer)
library(edgeR)
library(limma)
library(Biobase)
library(SummarizedExperiment)
library(tidyverse) 
library(ggfortify)
library(cluster)
library(edgeR)
library(limma)
library(Homo.sapiens)
library(BiocParallel)
library(qvalue)
library(pheatmap)
library(clusterProfiler)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(RColorBrewer)
library(variancePartition)
library(DOSE)
library(UpSetR)
library(ggvenn)
library(biomaRt)
library(ggridges)
library(reshape2)
library(BioNERO)
library(WGCNA)
library(impute)
library(dynamicTreeCut)
library(ReactomePA)
library(scales)

```


# Read in Data 
```{r Read in data, include=FALSE}

# 1. RUVg Corrected data across all 10 samples that has been log2 transformed. It has not yet been quantile normalized. 
 RUVg_Log2_quantnormalized_all10samples <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/RUVIII_10samples_log2_notquantilenormalized.csv", header = TRUE, row.names = 1)
RUVg_Log2_quantnormalized_all10samples <- RUVg_Log2_quantnormalized_all10samples^2
 

# 2. Toptable correspondding to the Diff.Abundance test from #1 
toptable_summary <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/Toptable_summary_RUVIII.csv", header = TRUE, row.names = 1)


# 3. Meta data for all 10 samples in our study. 
Meta <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/Meta.csv", header = TRUE, row.names = 1)


Toptable_Modules <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/SUPP_Table_2.csv", header = TRUE)

colnames(Toptable_Modules) <- c("X", "Protein","kTotal", "kWithin",        "kOut", "kDiff", "logFC", "AveExpr", "t", "P.Value","adj.P.Val" ,"B" , "threshold_P", "Modules", "DE_or_Not", "Norm_kIN", "Norm_kOut", "logFC.y",     "AveExpr.y", "t.y", "P.Value.y", "adj.P.Val.y", "B.y" , "threshold_P.y",    "Modules.y", "DE_or_Not.y", "Is_DA", "Is_DOXcorrelated", "Is_Hub", "Is_Cis_pQTL", "Is_Trans_pQTL", "Is_pQTL", "pLI_assigned", "pLI_Mut.Intolerant", "pLI_Mut.Tolerant", "Is_Druggable", "Is_CVD_protein",    "Is_CVD_PPI_protein")

New_RNA_PRO_DF_2 <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Data_Frames/DIA_proteins/RUVIII_Imputed/New_RNA_PRO_DF.csv", header = TRUE, row.names = 1)


New_RNA_PRO_DF_3 <- merge(Toptable_Modules, New_RNA_PRO_DF_2, by.x = "Protein", by.y = "uniprotswissprot")


hubs <- read.csv( file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/hubs.csv", header = TRUE)


GO_results_DOX <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/GO_results.csv", header = TRUE)


HPA_General3 <- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/HPA_General3_test.RData")

HPA_General4_test <- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/HPA_General4.RData")

HPA_General2_test <- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/HPA_General2.RData")


dbd_uniprot_list_test<- readRDS("/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/dbd_uniprot_list.RData")


RBP_pros_uniprot_3 <- read.csv(file = "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/RBP_pros_uniprot_3.csv", header = TRUE)

# Set the path to the folder containing the .tsv files
folder_path <- "/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/HPA_Metabolism" 

EnzymeTable <- read.csv(file ="/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/HPA_Metabolism_Labels/HPA_Metabo_labels.csv" , header = TRUE)

TF_UNIPROT_ENSEMBL_2_2 <- read.csv(file ="/Users/omarjohnson/Documents/Projects/Dox_Proteomics/Data/Proteomics/Data_sets/Supplement/TF_UNIPROT_ENSEMBL_2.csv" , header = TRUE)

```


# Functions 
```{r Functions}
# 1 
perform_module_comparisons_mutexc <- function(df, module_col, value_col) {
  # Ensure the necessary columns exist
  if (!(module_col %in% names(df) && value_col %in% names(df))) {
    stop("Specified columns do not exist in the dataframe.")
  }

  # Get a list of all unique modules
  modules <- unique(df[[module_col]])

  # Initialize an empty dataframe to store results
  results <- data.frame(Module1 = character(),
                        Module2 = character(),
                        WilcoxPValue = numeric(),
                        stringsAsFactors = FALSE)

  # Loop through each module
  for (module in modules) {
    # Data for the current module
    current_data <- df %>% filter(!!sym(module_col) == module) %>% pull(!!sym(value_col))

    # Data for all other modules
    other_data <- df %>% filter(!!sym(module_col) != module) %>% pull(!!sym(value_col))

    # Perform the Wilcoxon test
    test_result <- wilcox.test(current_data, other_data)

    # Add the results to the dataframe
    results <- rbind(results, data.frame(Module1 = module,
                                         Module2 = "Others",
                                         WilcoxPValue = test_result$p.value))
  }

  return(results)
}


# 2 
perform_module_comparisons_mutexc_2 <- function(df, module_col, value_col) {
  # Ensure the necessary columns exist
  if (!(module_col %in% names(df) && value_col %in% names(df))) {
    stop("Specified columns do not exist in the dataframe.")
  }

  # Get a list of all unique modules
  modules <- unique(df[[module_col]])

  # Initialize an empty list to store combined data frames
  combined_df_list <- list()

  # Initialize an empty dataframe to store results
  results <- data.frame(Module1 = character(),
                        Module2 = character(),
                        WilcoxPValue = numeric(),
                        stringsAsFactors = FALSE)

  # Loop through each module
  for (module in modules) {
    # Data for the current module
    current_data <- df %>% filter(!!sym(module_col) == module) %>% 
                    mutate(Group = as.character(module))

    # Data for all other modules
    other_data <- df %>% filter(!!sym(module_col) != module) %>% 
                    mutate(Group = paste("Not", module, sep=""))

    # Combine current module data with other module data
    combined_data <- rbind(current_data, other_data)

    # Add the combined data to the list
    combined_df_list[[module]] <- combined_data

    # Perform the Wilcoxon test
    test_result <- wilcox.test(current_data[[value_col]], other_data[[value_col]])

    # Add the results to the dataframe
    results <- rbind(results, data.frame(Module1 = module,
                                         Module2 = "Others",
                                         WilcoxPValue = test_result$p.value))
  }

  return(list("results" = results, "combined_data" = combined_df_list))
}


# 3 
# Generate enrichment function limited to the EXPRESSED proteins.
perform_module_disease_analysis_2 <- function(toptable, diseaseGenes) {
  # Prepare an empty list to collect results
  results <- list()
  
  # Ensure 'Modules' and 'Protein' columns exist in 'toptable'
  if(!"Modules" %in% names(toptable)) {
    stop("Column 'Modules' not found in the 'toptable'.")
  }
  if(!"Protein" %in% names(toptable)) {
    stop("Column 'Protein' not found in the 'toptable'.")
  }
  
  # Filter disease genes to include only those that are expressed in toptable
  expressedDiseaseGenes <- lapply(diseaseGenes, function(genes) {
    intersect(genes, toptable$Protein)
  })
  
  # Loop through each module
  modules <- unique(toptable$Modules)
  for (module in modules) {
    # Get the proteins in the module
    moduleGenes <- toptable$Protein[toptable$Modules == module]
    
    # Loop through each disease gene set
    for (diseaseName in names(expressedDiseaseGenes)) {
      # Find the intersecting genes between the module and the expressed disease genes
      diseaseModuleIntersect <- intersect(moduleGenes, expressedDiseaseGenes[[diseaseName]])
      
      # Calculate elements for the contingency table
      numIntersect = length(diseaseModuleIntersect)
      numInModuleNotDisease = length(moduleGenes) - numIntersect
      numInDiseaseNotModule = length(expressedDiseaseGenes[[diseaseName]]) - numIntersect
      numInNeither = nrow(toptable) - (numIntersect + numInModuleNotDisease + numInDiseaseNotModule)
      
      # Build the contingency table
      table <- matrix(c(
        numIntersect, # Both in disease list and module
        numInModuleNotDisease, # In module but not disease list
        numInDiseaseNotModule, # In disease list but not module
        numInNeither # In neither list
      ), nrow = 2, byrow = TRUE)
      
      # Perform chi-squared test and Fisher's exact test with error handling
      chiSqTestResult <- tryCatch({
        chisq.test(table, correct = TRUE)
      }, error = function(e) {
        list(p.value = NA)
      }, warning = function(w) {
        list(p.value = NA)
      })
      
      fisherTestResult <- tryCatch({
        fisher.test(table)
      }, error = function(e) {
        list(p.value = NA, odds.ratio = NA)
      }, warning = function(w) {
        list(p.value = NA, odds.ratio = NA)
      })
      
      # Calculate percent overlap, handle division by zero
      percentOverlap <- if (length(moduleGenes) > 0) {
        (numIntersect / length(expressedDiseaseGenes[[diseaseName]])) * 100
      } else {
        0
      }
      
      # Convert intersecting genes to a single character string
      intersectingGenesStr <- if (numIntersect > 0) {
        paste(diseaseModuleIntersect, collapse = ";")
      } else {
        ""  # Use an empty string to indicate no intersection
      }
      
      # Append to results list
      results[[paste(module, diseaseName, sep = "_")]] <- data.frame(
        Modules = module,
        Disease = diseaseName,
        ChiSqPValue = chiSqTestResult$p.value,
        FisherPValue = fisherTestResult$p.value,
        OddsRatio = fisherTestResult$estimate,
        PercentOverlap = percentOverlap,
        IntersectingGenes = intersectingGenesStr
      )
    }
  }
  
  # Combine results into a single data frame
  combined_df <- do.call(rbind, results)
  return(combined_df)
}


# 4 
# For testing for enrichment in the circos plot 
perform_module_disease_analysis <- function(toptable, diseaseGenes) {
  # Prepare an empty list to collect results
  results <- list()
  
  # Ensure 'Modules' and 'Protein' columns exist in 'toptable'
  if(!"Modules" %in% names(toptable)) {
    stop("Column 'Modules' not found in the 'toptable'.")
  }
  if(!"Protein" %in% names(toptable)) {
    stop("Column 'Protein' not found in the 'toptable'.")
  }
  
  # Filter disease genes to include only those that are expressed in toptable
  expressedDiseaseGenes <- lapply(diseaseGenes, function(genes) {
    intersect(genes, toptable$Protein)
  })
  
  # Loop through each module
  modules <- unique(toptable$Modules)
  for (module in modules) {
    # Get the genes in the module
    moduleGenes <- toptable$Protein[toptable$Modules == module]
    
    # Loop through each disease gene set
    for (diseaseName in names(expressedDiseaseGenes)) {
      # Find the intersecting genes between the module and the expressed disease genes
      diseaseModuleIntersect <- intersect(moduleGenes, expressedDiseaseGenes[[diseaseName]])
      
      # Calculate elements for the contingency table
      numIntersect = length(diseaseModuleIntersect)
      numInModuleNotDisease = length(moduleGenes) - numIntersect
      numInDiseaseNotModule = length(expressedDiseaseGenes[[diseaseName]]) - numIntersect
      numInNeither = nrow(toptable) - (numIntersect + numInModuleNotDisease + numInDiseaseNotModule)
      
      # Build the contingency table
      table <- matrix(c(
        numIntersect, # Both in disease list and module
        numInModuleNotDisease, # In module but not disease list
        numInDiseaseNotModule, # In disease list but not module
        numInNeither # In neither list
      ), nrow = 2, byrow = TRUE)
      
      # Perform chi-squared test and Fisher's exact test with error handling
      chiSqTestResult <- tryCatch({
        chisq.test(table, correct = TRUE)
      }, error = function(e) {
        list(p.value = NA)
      }, warning = function(w) {
        list(p.value = NA)
      })
      
      fisherTestResult <- tryCatch({
        fisher.test(table)
      }, error = function(e) {
        list(p.value = NA)
      }, warning = function(w) {
        list(p.value = NA)
      })
      
      # Calculate percent overlap, handle division by zero
      percentOverlap <- if (length(moduleGenes) > 0) {
        (numIntersect / length(expressedDiseaseGenes[[diseaseName]])) * 100
      } else {
        0
      }
      
      # Convert intersecting genes to a single character string
      intersectingGenesStr <- if (numIntersect > 0) {
        paste(diseaseModuleIntersect, collapse = ";")
      } else {
        ""  # Use an empty string to indicate no intersection
      }
      
      # Append to results list
      results[[paste(module, diseaseName, sep = "_")]] <- data.frame(
        Modules = module,
        Disease = diseaseName,
        ChiSqPValue = chiSqTestResult$p.value,
        FisherPValue = fisherTestResult$p.value,
        PercentOverlap = percentOverlap,
        IntersectingGenes = intersectingGenesStr
      )
    }
  }
  
  # Combine results into a single data frame
  results_df <- do.call(rbind, results)
  return(results_df)
}
```


# Fig-4A & S9 Biological process enrichment
```{r Modified GO representation 2}
GO_results_DOX %>% head()
GO_results_DOX$ONTOLOGY %>% unique()
GO_results_DOX$Module %>% unique()


# Convert the data frame to a list
list_input <- GO_results_DOX %>%
  group_by(Module) %>%
  summarise(Description = list(Description)) %>%
  deframe()


# Upset plot
upset(
  fromList(list_input),
  sets = names(list_input), # show all sets in the plot
  order.by = "freq",
  sets.bar.color = "#56B4E9",
  matrix.color = "black",
  main.bar.color = "#D55E00"
)




GO_results_DOX$p.adj.log <- -log(GO_results_DOX$p.adjust)
GO_results_DOX_green <- GO_results_DOX[GO_results_DOX$Module == "green", ]

GO_results_DOX_green_sorted <- GO_results_DOX_green[order(GO_results_DOX_green$p.adj.log), ]

GO_results_DOX_green_sorted$Description <-factor(GO_results_DOX_green_sorted$Description, levels = rev(GO_results_DOX_green_sorted$Description))


# 
# 
# GO_results_DOX_sorted <- GO_results_DOX[order(GO_results_DOX$p.adj.log), ]
# 
# GO_results_DOX_sorted$Description <-factor(GO_results_DOX_sorted$Description, levels = GO_results_DOX_sorted$Description)


# Plotting
 ggplot(GO_results_DOX_green_sorted, aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()+
   ylim(c(0,130))
 

 
 # Plotting
 ggplot(GO_results_DOX_green_sorted[150:nrow(GO_results_DOX_green_sorted),  ], aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()+
   ylim(c(0,130))
 
 
  # Plotting
 ggplot(GO_results_DOX_green_sorted[75:150,], aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()+
   ylim(c(0,130))
 
 
  # Plotting
 ggplot(GO_results_DOX_green_sorted[1:75, ], aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()+
   ylim(c(0,130))
 
 
 
 GO_results_DOX$p.adj.log <- -log(GO_results_DOX$p.adjust)
GO_results_DOX_green <- GO_results_DOX[GO_results_DOX$Module == "darkgreen", ]

GO_results_DOX_green_sorted <- GO_results_DOX_green[order(GO_results_DOX_green$p.adj.log), ]

GO_results_DOX_green_sorted$Description <-factor(GO_results_DOX_green_sorted$Description, levels = rev(GO_results_DOX_green_sorted$Description))


# Plotting
 ggplot(GO_results_DOX_green_sorted, aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()

 
 
 
 
 GO_results_DOX$p.adj.log <- -log(GO_results_DOX$p.adjust)
GO_results_DOX_green <- GO_results_DOX[GO_results_DOX$Module == "salmon", ]

GO_results_DOX_green_sorted <- GO_results_DOX_green[order(GO_results_DOX_green$p.adj.log), ]

GO_results_DOX_green_sorted$Description <-factor(GO_results_DOX_green_sorted$Description, levels = rev(GO_results_DOX_green_sorted$Description))


# Plotting
 ggplot(GO_results_DOX_green_sorted, aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()
 
 
 
 
 
 
 GO_results_DOX$p.adj.log <- -log(GO_results_DOX$p.adjust)
GO_results_DOX_green <- GO_results_DOX[GO_results_DOX$Module == "lightyellow", ]

GO_results_DOX_green_sorted <- GO_results_DOX_green[order(GO_results_DOX_green$p.adj.log), ]

GO_results_DOX_green_sorted$Description <-factor(GO_results_DOX_green_sorted$Description, levels = rev(GO_results_DOX_green_sorted$Description))


# Plotting
 ggplot(GO_results_DOX_green_sorted, aes(x = Description, y = p.adj.log, fill = Module)) +
  geom_bar(stat = "identity") +
  labs(title = "",
       x = "",
       y = "-log-adj. pval") +
  theme_dark() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
   scale_fill_identity()


```


# Fig-4B Enriched protein categories
```{r Enriched protein categories}
#### Main heatmap #### 
# Plot results 
results_df <- perform_module_disease_analysis_2(toptable = Toptable_Modules, diseaseGenes = HPA_General3)


melted_results <- melt(results_df, id.vars = c("Modules", "Disease", "PercentOverlap", "FisherPValue","OddsRatio" ))


# First, create a new column that indicates where to place stars
melted_results$Star <- ifelse(melted_results$FisherPValue < 0.05 & melted_results$OddsRatio > 1, "*", "")



melted_results_2 <- melted_results
melted_results_2 <- melted_results
module_order <- c("green","darkgreen","midnightblue","salmon","lightyellow", "lightgreen","blue", "magenta","darkred", "brown", "yellow", "royalblue", "grey")

# Factor the Module column in Fulltrait_df
melted_results_2$Modules <- factor(melted_results_2$Modules, levels = module_order)

melted_results_3 <- melted_results_2[melted_results_2$Modules %in% c("green", "darkgreen", "midnightblue", "salmon","lightyellow"), ]

# Vertical
ggplot(melted_results_3, aes(x = Modules, y = Disease, fill = OddsRatio)) +
  geom_tile(color = "black") +  
  scale_fill_gradientn(colors = c("white", "darkviolet", "red"), 
                       values = scales::rescale(c(0, 3, 6)), 
                       na.value = "grey50", name = "Odds ratio") +
  geom_text(aes(label = Star), color = "black", size = 8, na.rm = TRUE) +
  labs(x = "", y = "") +  
  theme_minimal() +  
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    axis.text.y = element_text(hjust = 0.5),
    axis.title = element_text(size = 12),
    legend.key.size = unit(0.5, 'cm'),
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )




# Subsetting the graph 
ggplot(melted_results_2, aes(x = Modules, y = Disease, fill = OddsRatio)) +
  geom_tile(color = "black") +  
  scale_fill_gradientn(colors = c("white", "darkviolet", "red"), 
                       values = scales::rescale(c(0, 2, 30)), 
                       na.value = "grey50", name = "Odds ratio") +
  geom_text(aes(label = Star), color = "black", size = 8, na.rm = TRUE) +
  labs(x = "", y = "") +  
  theme_minimal() +  
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    axis.text.y = element_text(hjust = 0.5),
    axis.title = element_text(size = 12),
    legend.key.size = unit(1, 'cm'),
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )

melted_results_3_sub <- melted_results_3[melted_results_3$Disease %in% c("transporters", "transcription factors", "stress granules", "RNA binding proteins", "enzymes"), ]




ggplot(melted_results_3_sub, aes(x = Modules, y = Disease, fill = OddsRatio)) +
  geom_tile(color = "black") +  
  scale_fill_gradientn(colors = c("white", "darkviolet", "red"), 
                       values = scales::rescale(c(0, 3, 6)), 
                       na.value = "grey50", name = "Odds ratio") +
  geom_text(aes(label = Star), color = "black", size = 8, na.rm = TRUE) +
  labs(x = "", y = "") +  
  theme_minimal() +  
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    axis.text.y = element_text(hjust = 0.5),
    axis.title = element_text(size = 12),
    legend.key.size = unit(0.5, 'cm'),
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )



#### Barplots #### 
Combined_for_barplots <- HPA_General3
Combined_for_barplots$RBPs <-HPA_General4_test$`All RBP types`
Combined_for_barplots$TFs <-TF_UNIPROT_ENSEMBL_2_2$uniprotswissprot 





# Get resultsdf:
results_df <- perform_module_disease_analysis_2(toptable = Toptable_Modules, diseaseGenes = Combined_for_barplots)


# TFs: 
results_df_DOXcorr <- results_df[results_df$Modules %in% c("green"), ]

results_df_DOXcorr_trans <- results_df_DOXcorr[results_df_DOXcorr$Disease == "TFs", ]


results_df_DOXcorr_Secreted <- c(strsplit(results_df_DOXcorr_trans$IntersectingGenes, ";")) %>% unlist()


results_df_DOXcorr_Secreted_hub <- results_df_DOXcorr_Secreted

TF_Dataforplot <- New_RNA_PRO_DF_3[New_RNA_PRO_DF_3$Protein %in% results_df_DOXcorr_Secreted_hub, ]


# Order the data frame by the absolute values of logFC in descending order
TF_Dataforplot_ordered <- TF_Dataforplot[order(-abs(TF_Dataforplot$logFC)), ]

# Subset the top 20 rows
TF_Dataforplot_top_20 <- head(TF_Dataforplot_ordered, 20)

# ------ Adjusted code for barplot for transcription factors 
# Order the data frame by the absolute values of logFC in descending order
TF_Dataforplot_ordered <- TF_Dataforplot[order(-abs(TF_Dataforplot$Pro_LogFC)), ]

# Subset the top 20 rows
TF_Dataforplot_top_20 <- head(TF_Dataforplot_ordered, 20)

# Reorder hgnc_symbol based on Pro_LogFC
TF_Dataforplot_top_20$hgnc_symbol <- factor(TF_Dataforplot_top_20$hgnc_symbol, 
                                            levels = TF_Dataforplot_top_20$hgnc_symbol[order(TF_Dataforplot_top_20$Pro_LogFC)])

# Plot
ggplot(TF_Dataforplot_top_20, aes(x = hgnc_symbol, y = Pro_LogFC, fill = Modules)) +
  geom_bar(stat = "identity") +
  labs(
    y = "logFC",
    fill = "Modules"
  ) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_identity() +
  ylim(c(-3.5, 3.5))


HPA_General4_test <-HPA_General2_test

# RBPs: 
Combined_for_barplots$RBPs <- HPA_General4_test$`All RBP types`
results_df$Disease %>% unique()
results_df_DOXcorr <- results_df[results_df$Modules %in% c("green", "lightyellow"), ]

results_df_DOXcorr_trans <- results_df_DOXcorr[results_df_DOXcorr$Disease == "RBPs", ]
results_df_DOXcorr_trans

results_df_DOXcorr_Secreted <- c(strsplit(results_df_DOXcorr_trans$IntersectingGenes, ";")[[1]], 
           strsplit(results_df_DOXcorr_trans$IntersectingGenes, ";")[[2]])

results_df_DOXcorr_Secreted_hub <- results_df_DOXcorr_Secreted

TF_Dataforplot <- New_RNA_PRO_DF_3[New_RNA_PRO_DF_3$Protein %in% results_df_DOXcorr_Secreted_hub, ]


# Order the data frame by the absolute values of logFC in descending order
TF_Dataforplot_ordered <- TF_Dataforplot[order(-abs(TF_Dataforplot$logFC)), ]

# Subset the top 20 rows
TF_Dataforplot_top_20 <- head(TF_Dataforplot_ordered, 20)

# ------ Adjusted code for barplot for transcription factors 
# Order the data frame by the absolute values of logFC in descending order
TF_Dataforplot_ordered <- TF_Dataforplot[order(-abs(TF_Dataforplot$Pro_LogFC)), ]

# Subset the top 20 rows
TF_Dataforplot_top_20 <- head(TF_Dataforplot_ordered, 20)

# Reorder hgnc_symbol based on Pro_LogFC
TF_Dataforplot_top_20$hgnc_symbol <- factor(TF_Dataforplot_top_20$hgnc_symbol, 
                                            levels = TF_Dataforplot_top_20$hgnc_symbol[order(TF_Dataforplot_top_20$Pro_LogFC)])

# Plot
ggplot(TF_Dataforplot_top_20, aes(x = hgnc_symbol, y = Pro_LogFC, fill = Modules)) +
  geom_bar(stat = "identity") +
  labs(
    y = "logFC",
    fill = "Modules"
  ) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_identity() +
  ylim(c(-3.5, 3.5))








# ENZYMES: 
results_df_DOXcorr <- results_df[results_df$Modules %in% c("darkgreen", "salmon"), ]

results_df_DOXcorr$Disease
results_df_DOXcorr_trans <- results_df_DOXcorr[results_df_DOXcorr$Disease == "enzymes", ]

results_df_DOXcorr_Secreted <- c(strsplit(results_df_DOXcorr_trans$IntersectingGenes, ";")[[1]], 
           strsplit(results_df_DOXcorr_trans$IntersectingGenes, ";")[[2]])

results_df_DOXcorr_Secreted_hub <- results_df_DOXcorr_Secreted

TF_Dataforplot <- New_RNA_PRO_DF_3[New_RNA_PRO_DF_3$Protein %in% results_df_DOXcorr_Secreted_hub, ]


# Order the data frame by the absolute values of logFC in descending order
TF_Dataforplot_ordered <- TF_Dataforplot[order(-abs(TF_Dataforplot$logFC)), ]

# Subset the top 20 rows
TF_Dataforplot_top_20 <- head(TF_Dataforplot_ordered, 20)

# ------ Adjusted code for barplot for transcription factors 
# Order the data frame by the absolute values of logFC in descending order
TF_Dataforplot_ordered <- TF_Dataforplot[order(-abs(TF_Dataforplot$Pro_LogFC)), ]

# Subset the top 20 rows
TF_Dataforplot_top_20 <- head(TF_Dataforplot_ordered, 20)

# Reorder hgnc_symbol based on Pro_LogFC
TF_Dataforplot_top_20$hgnc_symbol <- factor(TF_Dataforplot_top_20$hgnc_symbol, 
                                            levels = TF_Dataforplot_top_20$hgnc_symbol[order(TF_Dataforplot_top_20$Pro_LogFC)])

# Plot for all enzymes
ggplot(TF_Dataforplot_top_20, aes(x = hgnc_symbol, y = Pro_LogFC, fill = Modules)) +
  geom_bar(stat = "identity") +
  labs(
    y = "logFC",
    fill = "Modules"
  ) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_identity() +
  ylim(c(-3.5, 3.5))

TF_Dataforplot_top_20 %>% dim()


```


# Fig-4C Enriched TF categories
```{r Enriched TF factor DBD}
#### Heatmap #### 
# Plot results 
results_df <- perform_module_disease_analysis_2(toptable = Toptable_Modules, diseaseGenes = dbd_uniprot_list_test)


results_df[(results_df$FisherPValue < 0.05) & (results_df$Modules %in% c("green","darkgreen","midnightblue","salmon","lightyellow")), ]


melted_results <- melt(results_df, id.vars = c("Modules", "Disease", "PercentOverlap", "FisherPValue","OddsRatio" ))


# First, create a new column that indicates where to place stars
melted_results$Star <- ifelse(melted_results$FisherPValue < 0.05 & melted_results$OddsRatio > 1, "*", "")


melted_results_2 <- melted_results


melted_results_2 <- melted_results
module_order <- c("green","darkgreen","midnightblue","salmon","lightyellow", "lightgreen","blue", "magenta","darkred", "brown", "yellow", "royalblue", "grey")

# Factor the Module column in Fulltrait_df
melted_results_2$Modules <- factor(melted_results_2$Modules, levels = module_order)


melted_results_3 <- melted_results_2
melted_results_4 <- melted_results_3[melted_results_3$Modules %in% c("green", "darkgreen", "midnightblue", "salmon", "lightyellow"), ]

melted_results_4$Disease %>% unique()

melted_results_5 <- melted_results_4[melted_results_4$Disease %in% c("Homeodomain","GATA","C2H2 ZF"), ]



ggplot(melted_results_5, aes(x = Modules, y = Disease, fill = OddsRatio)) +
  geom_tile(color = "black") +  
  scale_fill_gradientn(colors = c("white", "darkviolet", "red"), 
                       values = scales::rescale(c(0, 5, 50)), 
                       na.value = "grey50", name = "Odds ratio") +
  geom_text(aes(label = Star), color = "black", size = 6, na.rm = TRUE) +
  labs(x = "", y = "") +  
  theme_minimal() +  
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    axis.text.y = element_text(hjust = 0.5),
    axis.title = element_text(size = 12),
    legend.key.size = unit(0.6, 'cm'),
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )









#### Barplots #### 
results_df_DOXcorr <- results_df[results_df$Modules == "green", ]


results_df_DOXcorr_Secreted <- strsplit(results_df_DOXcorr$IntersectingGenes, ";") %>% unlist()

results_df_DOXcorr_Secreted_hub <- results_df_DOXcorr_Secreted


ggplot(New_RNA_PRO_DF_3[New_RNA_PRO_DF_3$Protein %in% results_df_DOXcorr_Secreted_hub, ], aes(x = hgnc_symbol, y = Pro_LogFC, fill = Modules)) +
  geom_bar(stat = "identity") +
  labs(
    x = "Protein",
    y = "logFC",
    fill = "Modules",
    title = "DOX corr. transcrption factors"
  ) +
  theme_dark() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  scale_fill_identity()+
  ylim(c(-3, 3))
```


# Fig-4D Enriched RBP categories
```{r Enriched RBP categories}
RBP_pros_uniprot_3 %>% head()
 
# RBPs all types
HPA_General2_test$RBP_All_types <- RBP_pros_uniprot_3$uniprotswissprot


# RBPs thaat are DNA Binding 
HPA_General2_test$RBP_DNA_Binding <-
RBP_pros_uniprot_3[RBP_pros_uniprot_3$ChIPseq.HepG1 == 1|RBP_pros_uniprot_3$ChIPseq.K561 == 1 , ]$uniprotswissprot


# RBPs that are essential
HPA_General2_test$RBP_Essential <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$Essential.Genes == 1, ]$uniprotswissprot


# RBP Splicing.regulation
HPA_General2_test$RBP_Splicing_regulation <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$Splicing.regulation == 1, ]$uniprotswissprot


# RBP spliceosome
HPA_General2_test$RBP_spliceosome <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$Spliceosome == 1, ]$uniprotswissprot


# RBP RNA modifyinig
HPA_General2_test$RBP_RNA_modifynig <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$RNA.modification == 1, ]$uniprotswissprot


# RBP RNA modifyinig
HPA_General2_test$RBP_3UTR_processing <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$X3..end.processing == 1, ]$uniprotswissprot


# RBP RNA modifyinig
HPA_General2_test$RBP_rRNA_processing <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$rRNA.processing == 1, ]$uniprotswissprot


# RBP Ribosome translation 
HPA_General2_test$RBP_Ribosome_translation <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$Ribosome...basic.translation == 1, ]$uniprotswissprot


# RBP mRNA_stability
HPA_General2_test$RBP_mRNA_stability <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$RNA.stability...decay == 1, ]$uniprotswissprot


# RBP microRNA_processing 
HPA_General2_test$RBP_microRNA_processing <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$microRNA.processing == 1, ]$uniprotswissprot


# RBP RNA.localization 
HPA_General2_test$RBP_RNA_localization <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$RNA.localization == 1, ]$uniprotswissprot


# RBP RNA.export 
HPA_General2_test$RBP_RNA_export <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$RNA.export == 1, ]$uniprotswissprot


# RBP_Translation_reg
HPA_General2_test$RBP_Translation_reg <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$Translation.regulation == 1, ]$uniprotswissprot


# RBP tRNA.regulation
HPA_General2_test$RBP_tRNA.regulation <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$tRNA.regulation == 1, ]$uniprotswissprot


# RBP_mitochond._RNA_reg
HPA_General2_test$RBP_mitochond._RNA_reg <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$mitochondrial.RNA.regulation == 1, ]$uniprotswissprot


# RBP_Stress_granules
HPA_General2_test$RBP_Stress_granules <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$P.body...stress.granules == 1, ]$uniprotswissprot


# RBP_Exon.Junction.Complex
HPA_General2_test$RBP_Exon.Junction.Complex <- RBP_pros_uniprot_3[RBP_pros_uniprot_3$Exon.Junction.Complex == 1, ]$uniprotswissprot


HPA_General4_test <-HPA_General2_test


HPA_General4_test <- HPA_General2_test[5:length(HPA_General2_test)]

names(HPA_General4_test) <- c("DNA Binding" ,"Essential", "splicing regulation","spliceosome", "RNA_modifynig", "3'UTR processing", "rRNA_processing", "Ribosome-translation", "mRNA_stability", "microRNA processing","RNA localization", "RNA_export", "Translation regulation", "tRNA regulation", "mitochondrial RNA regulation", "stress granules", "exon junction complex", "All RBP types")




results_df <- perform_module_disease_analysis_2(toptable = Toptable_Modules, diseaseGenes = HPA_General4_test)


melted_results <- melt(results_df, id.vars = c("Modules", "Disease", "PercentOverlap", "FisherPValue","OddsRatio" ))


# First, create a new column that indicates where to place stars
melted_results$Star <- ifelse(melted_results$FisherPValue < 0.05 & melted_results$OddsRatio > 1, "*", "")


melted_results_2 <- melted_results

melted_results_2 <- melted_results
module_order <- c("green","darkgreen","midnightblue","salmon","lightyellow", "lightgreen","blue", "magenta","darkred", "brown", "yellow", "royalblue", "grey")

# Factor the Module column in Fulltrait_df
melted_results_2$Modules <- factor(melted_results_2$Modules, levels = module_order)

melted_results_3 <- melted_results_2[melted_results_2$Modules %in% c("green","darkgreen","midnightblue","salmon","lightyellow"), ]
thingstosubset <- melted_results_3[(melted_results_3$FisherPValue< 0.05)& (melted_results_3$OddsRatio > 1), ]$Disease
thingstosubset

melted_results_4 <- melted_results_3[melted_results_3$Disease %in% thingstosubset, ]


# Vertical
ggplot(melted_results_4, aes(x = Modules, y = Disease, fill = OddsRatio)) +
  geom_tile(color = "black") +  
  scale_fill_gradientn(colors = c("white", "darkviolet", "red"), 
                       values = scales::rescale(c(0, 3, 100)), 
                       na.value = "grey50", name = "Odds ratio") +
  geom_text(aes(label = Star), color = "black", size = 4, na.rm = TRUE) +
  labs(x = "", y = "") +  
  theme_minimal() +  
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    axis.text.y = element_text(hjust = 0.5),
    axis.title = element_text(size = 12),
    legend.key.size = unit(0.6, 'cm'),
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )


```


# Fig-4E Enriched Enzyme categories
```{r Enriched Enzyme categories}
# Get the list of .tsv files
file_names <- list.files(path = folder_path, pattern = "\\.tsv$", full.names = TRUE)



# Read each file and extract the 'Uniprot' column
HPA_Metabo <- lapply(file_names, function(file) {
  data <- read.table(file, header = TRUE, sep = "\t") # for tab-separated values
  return(data$Uniprot) # 'Uniprot' is column name
})


# Name the list elements without the full path and extension
names(HPA_Metabo) <- sapply(strsplit(basename(file_names), "\\."), `[`, 1)
EnzymeClassNamesTosave <- names(HPA_Metabo)


EnzymeTable %>% head()
colnames(EnzymeTable) <- c("ID", "Filename", "Description")
EnzymeTable %>% head()
names(HPA_Metabo) <- EnzymeTable$Description

HPA_Metabo$`Beta oxidation of fatty acids` <- unique(unlist(c(HPA_Metabo[10:21])))
HPA_Metabo$`Carnitine shuttle` <- unique(unlist(c(HPA_Metabo[31:33])))
HPA_Metabo$`Fatty acid activaton` <- unique(unlist(c(HPA_Metabo[46:47])))
HPA_Metabo$`Fatty acid biosynthesis` <- unique(unlist(c(HPA_Metabo[49:51])))


HPA_Metabo2 <- HPA_Metabo[c(1,2,3,4,5,135,136,137,22,37,40,41,42,48,56,60,62,63,66,70,92,95,103,104,105,106,107,108,109,111,113)]


# Plot results 
results_df <- perform_module_disease_analysis_2(toptable = Toptable_Modules, diseaseGenes = HPA_Metabo2)


melted_results <- melt(results_df, id.vars = c("Modules", "Disease", "PercentOverlap", "FisherPValue","OddsRatio" ))


# First, create a new column that indicates where to place stars
melted_results$Star <- ifelse(melted_results$FisherPValue < 0.05 & melted_results$OddsRatio > 1, "*", "")


melted_results_2 <- melted_results
module_order <- c("green","darkgreen","midnightblue","salmon","lightyellow", "lightgreen","blue", "magenta","darkred", "brown", "yellow", "royalblue", "grey")


# Factor the Module column in Fulltrait_df
melted_results_2$Modules <- factor(melted_results_2$Modules, levels = module_order)


Enzyme_termlistsubset <- melted_results_2[(melted_results_2$FisherPValue <0.05) & (melted_results_2$OddsRatio > 1) & (melted_results_2$Modules %in% c("green","darkgreen","midnightblue","salmon","lightyellow")), ]$Disease %>% unique()
Enzyme_termlistsubset

melted_results_3 <- melted_results_2[(melted_results_2$Disease %in% Enzyme_termlistsubset) & ((melted_results_2$Modules %in% c("green","darkgreen","midnightblue","salmon","lightyellow"))), ]
  

# Vertical
ggplot(melted_results_3, aes(x = Modules, y = Disease, fill = OddsRatio)) +
  geom_tile(color = "black") +  
  scale_fill_gradientn(colors = c("white", "darkviolet", "red"), 
                       values = scales::rescale(c(0, 3, 60)), 
                       na.value = "grey50", name = "Odds ratio") +
  geom_text(aes(label = Star), color = "black", size = 4, na.rm = TRUE) +
  labs(x = "", y = "") +  
  theme_minimal() +  
  theme(
    axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1),
    axis.text.y = element_text(hjust = 0.5),
    axis.title = element_text(size = 12),
    legend.key.size = unit(0.6, 'cm'),
    legend.title = element_text(size = 10),
    legend.text = element_text(size = 8)
  )


EZ_subset <- strsplit(melted_results_3$value, ";") %>% unlist() %>% unique()

EZ_Dataforplot_ordered <- TF_Dataforplot_ordered[TF_Dataforplot_ordered$Protein %in% EZ_subset, ]




EZ_Dataforplot_ordered <- EZ_Dataforplot_ordered[order(-abs(EZ_Dataforplot_ordered$Pro_LogFC)), ]

# Subset the top 20 rows
EZ_Dataforplot_ordered_top_20 <- head(EZ_Dataforplot_ordered, 20)

# Reorder hgnc_symbol based on Pro_LogFC
EZ_Dataforplot_ordered_top_20$hgnc_symbol <- factor(EZ_Dataforplot_ordered_top_20$hgnc_symbol, 
                                            levels = EZ_Dataforplot_ordered_top_20$hgnc_symbol[order(EZ_Dataforplot_ordered_top_20$Pro_LogFC)])

# Plot for all enzymes
ggplot(EZ_Dataforplot_ordered_top_20, aes(x = hgnc_symbol, y = Pro_LogFC, fill = Modules)) +
  geom_bar(stat = "identity") +
  labs(
    y = "logFC",
    fill = "Modules"
  ) +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  scale_fill_identity() +
  ylim(c(-3.5, 3.5))

```


